// @sansenjian/qq-music-api/dist/cli.cjs
#!/usr/bin/env node
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_cookieResolver = require("./cookieResolver-In4BnxUe.cjs");
const require_package = require("./package-8fvg49I3.cjs");
let node_fs = require("node:fs");
node_fs = require_cookieResolver.__toESM(node_fs, 1);
let node_path = require("node:path");
node_path = require_cookieResolver.__toESM(node_path, 1);
let node_url = require("node:url");
let node_os = require("node:os");
node_os = require_cookieResolver.__toESM(node_os, 1);
//#region src/cli.ts
var defaultIo = {
	stdout: (message) => console.log(message),
	stderr: (message) => console.error(message)
};
var resolveRealPath = (filePath) => {
	try {
		return node_fs.default.realpathSync.native(filePath);
	} catch {
		return node_path.default.resolve(filePath);
	}
};
var parsePort = (value) => {
	if (value === void 0 || value === "") throw new Error("Missing value for --port");
	if (!/^\d+$/.test(value)) throw new Error(`Invalid port: ${value}`);
	const port = Number(value);
	if (!Number.isInteger(port) || port < 0 || port > 65535) throw new Error(`Invalid port: ${value}`);
	return port;
};
var parseArgs = (argv) => {
	const args = [...argv];
	const parsed = {
		args: [],
		json: false,
		help: false,
		version: false
	};
	for (let index = 0; index < args.length; index += 1) {
		const arg = args[index];
		if (arg === "--json") {
			parsed.json = true;
			continue;
		}
		if (arg === "--help" || arg === "-h") {
			parsed.help = true;
			continue;
		}
		if (arg === "--version" || arg === "-v") {
			parsed.version = true;
			continue;
		}
		if (arg === "--port" || arg === "-p") {
			parsed.port = parsePort(args[index + 1]);
			index += 1;
			continue;
		}
		if (arg.startsWith("--port=")) {
			parsed.port = parsePort(arg.slice(7));
			continue;
		}
		parsed.args.push(arg);
	}
	return parsed;
};
var printJson = (io, payload) => {
	io.stdout(JSON.stringify(payload, null, 2));
};
var printError = (io, json, code, message) => {
	if (json) io.stderr(JSON.stringify({
		ok: false,
		error: {
			code,
			message
		}
	}));
	else io.stderr(`Error: ${message}`);
	return 1;
};
var helpText = () => `QQ Music API CLI

Usage:
  qq-music-api [serve] [--port <port>] [--json]
  qq-music-api config path [--json]
  qq-music-api config doctor [--json]
  qq-music-api doctor [--json]
  qq-music-api auth status [--json]
  qq-music-api auth clear [--json]

Options:
  --json             Output stable machine-readable JSON for supported commands.
  --port, -p <port>  Port for the serve command.
  --version, -v      Print package version.
  --help, -h         Show this help.

Notes:
  Running qq-music-api with no command keeps the legacy behavior and starts the HTTP service.
  Auth commands never print the full Cookie value.
  MCP is published separately as @sansenjian/qq-music-api-mcp.
`;
var getPathPayload = () => {
	return {
		ok: true,
		command: "config path",
		configDir: require_cookieResolver.getConfigDir(),
		serviceConfigPath: require_cookieResolver.resolveConfigPath("service-config.json"),
		userInfoPath: require_cookieResolver.resolveConfigPath("user-info.json")
	};
};
var readJsonFileStatus = (filePath) => {
	if (!node_fs.default.existsSync(filePath)) return {
		path: filePath,
		exists: false,
		status: "missing"
	};
	try {
		JSON.parse(node_fs.default.readFileSync(filePath, "utf-8"));
		return {
			path: filePath,
			exists: true,
			status: "ok"
		};
	} catch (error) {
		return {
			path: filePath,
			exists: true,
			status: "invalid",
			error: error instanceof Error ? error.message : "Invalid JSON"
		};
	}
};
var checkWritable = (configDir) => {
	try {
		node_fs.default.mkdirSync(configDir, { recursive: true });
		const probePath = node_path.default.join(configDir, `.doctor-${process.pid}-${Date.now()}.tmp`);
		node_fs.default.writeFileSync(probePath, "ok", "utf-8");
		node_fs.default.rmSync(probePath, { force: true });
		return { writable: true };
	} catch (error) {
		return {
			writable: false,
			error: error instanceof Error ? error.message : "Config directory is not writable"
		};
	}
};
var getDoctorPayload = () => {
	const configDir = require_cookieResolver.getConfigDir();
	const writable = checkWritable(configDir);
	const serviceConfig = readJsonFileStatus(require_cookieResolver.resolveConfigPath("service-config.json"));
	const userInfo = readJsonFileStatus(require_cookieResolver.resolveConfigPath("user-info.json"));
	const hasInvalidJson = serviceConfig.status === "invalid" || userInfo.status === "invalid";
	return {
		ok: writable.writable && !hasInvalidJson,
		command: "config doctor",
		configDir,
		platform: node_os.default.platform(),
		nodeVersion: process.version,
		writable,
		serviceConfig,
		userInfo
	};
};
var getAuthStatusPayload = () => {
	const userInfo = require_cookieResolver.getUserInfo();
	const cookieKeys = require_cookieResolver.getCookieKeys(userInfo.cookie);
	return {
		ok: true,
		command: "auth status",
		authenticated: Boolean(userInfo.cookie && (userInfo.uin || userInfo.loginUin)),
		uin: userInfo.uin || userInfo.loginUin || "",
		hasCookie: Boolean(userInfo.cookie),
		cookieKeys,
		cookieCount: cookieKeys.length
	};
};
var emptyUserInfo = () => ({
	loginUin: "",
	uin: "",
	cookie: "",
	cookieList: [],
	cookieObject: {},
	refreshData: () => ({})
});
var clearAuth = () => {
	const configDir = require_cookieResolver.getConfigDir();
	const userInfoPath = require_cookieResolver.resolveConfigPath("user-info.json");
	node_fs.default.mkdirSync(configDir, { recursive: true });
	node_fs.default.writeFileSync(userInfoPath, `${JSON.stringify({
		loginUin: "",
		cookie: ""
	}, null, 2)}\n`, "utf-8");
	require_cookieResolver.setUserInfo(emptyUserInfo());
	return {
		ok: true,
		command: "auth clear",
		cleared: true,
		userInfoPath
	};
};
var printHumanPath = (io) => {
	const payload = getPathPayload();
	io.stdout(`Config directory: ${payload.configDir}`);
	io.stdout(`Service config:    ${payload.serviceConfigPath}`);
	io.stdout(`User info:         ${payload.userInfoPath}`);
};
var printHumanDoctor = (io) => {
	const payload = getDoctorPayload();
	io.stdout(`Config directory: ${payload.configDir}`);
	io.stdout(`Writable:         ${payload.writable.writable ? "yes" : "no"}`);
	io.stdout(`service-config:   ${payload.serviceConfig.status}`);
	io.stdout(`user-info:        ${payload.userInfo.status}`);
	io.stdout(`Overall:          ${payload.ok ? "ok" : "needs attention"}`);
};
var printHumanAuthStatus = (io) => {
	const payload = getAuthStatusPayload();
	io.stdout(`Authenticated: ${payload.authenticated ? "yes" : "no"}`);
	io.stdout(`UIN:           ${payload.uin || "missing"}`);
	io.stdout(`Cookie:        ${payload.hasCookie ? "present" : "missing"}`);
	io.stdout(`Cookie keys:   ${payload.cookieKeys.length ? payload.cookieKeys.join(", ") : "none"}`);
};
var runCli = async (argv = process.argv.slice(2), io = defaultIo) => {
	let parsed;
	try {
		parsed = parseArgs(argv);
	} catch (error) {
		return printError(io, argv.includes("--json"), "INVALID_ARGS", error instanceof Error ? error.message : "Invalid arguments");
	}
	const [command = "serve", subcommand] = parsed.args;
	if (parsed.version) {
		io.stdout(require_package.package_default.version);
		return 0;
	}
	if (parsed.help) {
		io.stdout(helpText());
		return 0;
	}
	try {
		if (command === "serve") {
			const { startServer } = await Promise.resolve().then(() => require("./server-CdzMPnXx.cjs")).then((n) => n.server_exports);
			startServer({
				port: parsed.port,
				json: parsed.json
			});
			return 0;
		}
		if (command === "config" && subcommand === "path") {
			if (parsed.json) printJson(io, getPathPayload());
			else printHumanPath(io);
			return 0;
		}
		if (command === "config" && subcommand === "doctor" || command === "doctor") {
			if (parsed.json) printJson(io, getDoctorPayload());
			else printHumanDoctor(io);
			return 0;
		}
		if (command === "auth" && subcommand === "status") {
			if (parsed.json) printJson(io, getAuthStatusPayload());
			else printHumanAuthStatus(io);
			return 0;
		}
		if (command === "auth" && subcommand === "clear") {
			const payload = clearAuth();
			if (parsed.json) printJson(io, payload);
			else io.stdout(`Cleared auth state at ${payload.userInfoPath}`);
			return 0;
		}
		if (command === "mcp" && (subcommand === "start" || subcommand === void 0)) return printError(io, parsed.json, "MCP_PACKAGE_REQUIRED", "MCP support moved to @sansenjian/qq-music-api-mcp. Install it and run qq-music-api-mcp instead.");
		return printError(io, parsed.json, "UNKNOWN_COMMAND", `Unknown command: ${parsed.args.join(" ") || command}`);
	} catch (error) {
		return printError(io, parsed.json, "COMMAND_FAILED", error instanceof Error ? error.message : "Command failed");
	}
};
if ((process.argv[1] ? resolveRealPath(process.argv[1]) : "") === resolveRealPath((0, node_url.fileURLToPath)(require("url").pathToFileURL(__filename).href))) runCli().then((exitCode) => {
	process.exitCode = exitCode;
});
//#endregion
exports.runCli = runCli;
