// @sansenjian/qq-music-api/dist/cookieResolver-s0fxfqAN.js
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";
//#region src/util/logger.ts
var SENSITIVE_KEY_PATTERN = /(cookie|authorization|token|uin|qq|user)/i;
var shouldLog = (level) => {
	if (process.env.NODE_ENV === "test") return false;
	if (level === "debug") return process.env.DEBUG === "true";
	return true;
};
var maskSensitiveValue = (value) => {
	if (typeof value === "string") return value ? "[masked]" : value;
	if (Array.isArray(value)) return value.map(maskSensitiveValue);
	if (value && typeof value === "object") return sanitizeLogPayload(value);
	return value;
};
var sanitizeLogPayload = (payload) => {
	return Object.fromEntries(Object.entries(payload).map(([key, value]) => [key, SENSITIVE_KEY_PATTERN.test(key) ? maskSensitiveValue(value) : maskSensitiveValueIfNested(value)]));
};
var maskSensitiveValueIfNested = (value) => {
	if (Array.isArray(value)) return value.map(maskSensitiveValueIfNested);
	if (value && typeof value === "object") return sanitizeLogPayload(value);
	return value;
};
var writeLog = (level, message, payload) => {
	if (!shouldLog(level)) return;
	const args = payload ? [message, sanitizeLogPayload(payload)] : [message];
	if (level === "error") {
		console.error(...args);
		return;
	}
	if (level === "warn") {
		console.warn(...args);
		return;
	}
	console.log(...args);
};
var logger = {
	debug: (message, payload) => writeLog("debug", message, payload),
	info: (message, payload) => writeLog("info", message, payload),
	warn: (message, payload) => writeLog("warn", message, payload),
	error: (message, payload) => writeLog("error", message, payload)
};
//#endregion
//#region src/config/config-path.ts
var moduleDir = path.dirname(fileURLToPath(import.meta.url));
var configDirName = "config";
var distDirName = "dist";
var srcDirName = "src";
var resolvePackageRoot = (fromDir) => {
	const baseName = path.basename(fromDir);
	if (baseName === configDirName) {
		const parentDir = path.dirname(fromDir);
		if (path.basename(parentDir) === srcDirName) return path.dirname(parentDir);
		return path.basename(parentDir) === distDirName ? path.dirname(parentDir) : parentDir;
	}
	if (baseName === srcDirName) return path.dirname(fromDir);
	return baseName === distDirName ? path.dirname(fromDir) : fromDir;
};
var resolveDefaultConfigDir = (fromDir = moduleDir) => path.join(resolvePackageRoot(fromDir), configDirName);
var getConfigDir = () => {
	if (process.env.QQ_MUSIC_API_CONFIG_DIR) return path.resolve(process.env.QQ_MUSIC_API_CONFIG_DIR);
	return resolveDefaultConfigDir();
};
var resolveConfigPath = (filename) => path.join(getConfigDir(), filename);
//#endregion
//#region src/config/user-info.ts
var userInfo = {
	loginUin: "",
	cookie: "",
	cookieList: [],
	cookieObject: {},
	refreshData: () => ({})
};
var cookieList = [];
var cookieObject = {};
var infoPath = resolveConfigPath("user-info.json");
var ensureConfigDir = () => {
	fs.mkdirSync(path.dirname(infoPath), { recursive: true });
};
var parseCookieObject = (cookie) => {
	const parsed = {};
	cookie.split(";").map((item) => item.trim()).filter(Boolean).forEach((item) => {
		const separatorIndex = item.indexOf("=");
		if (separatorIndex <= 0) return;
		const key = item.slice(0, separatorIndex).trim();
		const value = item.slice(separatorIndex + 1).trim();
		if (key) parsed[key] = value;
	});
	return parsed;
};
var normalizeUserInfo = (value) => {
	const parsed = value && typeof value === "object" ? value : {};
	return {
		loginUin: typeof parsed.loginUin === "string" ? parsed.loginUin : "",
		cookie: typeof parsed.cookie === "string" ? parsed.cookie : "",
		cookieList: [],
		cookieObject: {},
		refreshData: () => ({})
	};
};
var initData = () => {
	try {
		userInfo = normalizeUserInfo(JSON.parse(fs.readFileSync(infoPath, "utf-8")));
	} catch {
		userInfo = {
			loginUin: "",
			cookie: "",
			cookieList: [],
			cookieObject: {},
			refreshData: () => ({})
		};
	}
	cookieList = (userInfo.cookie || "").split(";").map((item) => item.trim()).filter(Boolean);
	cookieObject = parseCookieObject(userInfo.cookie || "");
};
var refreshData = (cookie) => {
	const uinMatch = cookie.match(/(?:^|;\s*)uin=([^;]+)/);
	const uin = uinMatch ? uinMatch[1] : "";
	ensureConfigDir();
	fs.writeFileSync(infoPath, JSON.stringify({
		loginUin: uin,
		cookie
	}), "utf-8");
	initData();
	return {
		...userInfo,
		uin: userInfo.loginUin || cookieObject.uin,
		cookieList,
		cookieObject
	};
};
initData();
var exportObj = {
	loginUin: userInfo?.loginUin || "",
	uin: userInfo?.loginUin || cookieObject?.uin || "",
	cookie: userInfo?.cookie || "",
	cookieList: cookieList || [],
	cookieObject: cookieObject || {},
	refreshData
};
//#endregion
//#region src/config/user-info-store.ts
var createEmptyUserInfo = () => ({
	loginUin: "",
	uin: "",
	cookie: "",
	cookieList: [],
	cookieObject: {},
	refreshData: () => ({})
});
var setUserInfo = (userInfo) => {
	global.userInfo = userInfo || createEmptyUserInfo();
	return global.userInfo;
};
var initializeUserInfo = (userInfo = exportObj) => setUserInfo(userInfo);
var getUserInfo = () => {
	if (!global.userInfo) return initializeUserInfo();
	return global.userInfo;
};
var getUserCookie = () => getUserInfo().cookie || "";
var getUserUin = (fallback = "0") => {
	const userInfo = getUserInfo();
	return String(userInfo.uin || userInfo.loginUin || fallback);
};
//#endregion
//#region src/config/service-config.ts
var configPath = resolveConfigPath("service-config.json");
var defaultConfig = {
	fallbackMode: true,
	useGlobalCookie: false,
	cookieParamName: "cookie"
};
var isRecord = (value) => Boolean(value && typeof value === "object");
var normalizeServiceConfig = (value) => {
	if (!isRecord(value)) return defaultConfig;
	return {
		fallbackMode: typeof value.fallbackMode === "boolean" ? value.fallbackMode : defaultConfig.fallbackMode,
		useGlobalCookie: typeof value.useGlobalCookie === "boolean" ? value.useGlobalCookie : defaultConfig.useGlobalCookie,
		cookieParamName: typeof value.cookieParamName === "string" && value.cookieParamName.trim() ? value.cookieParamName.trim() : defaultConfig.cookieParamName
	};
};
var config = defaultConfig;
if (fs.existsSync(configPath)) try {
	const content = fs.readFileSync(configPath, "utf-8");
	config = normalizeServiceConfig(JSON.parse(content));
} catch {
	logger.warn("service-config.json 读取失败，使用默认配置");
}
if (process.env.FALLBACK_MODE === "true") config.fallbackMode = true;
if (process.env.USE_GLOBAL_COOKIE === "true") config.useGlobalCookie = true;
var service_config_default = config;
//#endregion
//#region src/util/cookieResolver.ts
var normalizeCookieValue = (value) => {
	if (Array.isArray(value)) return normalizeCookieValue(value[0]);
	if (typeof value !== "string") return;
	const normalized = value.trim();
	return normalized ? normalized : void 0;
};
var extractCookieValue = (cookie, name) => {
	if (!cookie) return void 0;
	const escapedName = name.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
	return cookie.match(new RegExp(`(?:^|;\\s*)${escapedName}=([^;]*)`))?.[1]?.trim() || void 0;
};
var getCookieKeys = (cookie) => {
	if (!cookie) return [];
	return cookie.split(";").map((item) => item.trim()).filter(Boolean).flatMap((item) => {
		const separatorIndex = item.indexOf("=");
		if (separatorIndex <= 0) return [];
		const key = item.slice(0, separatorIndex).trim();
		return key ? [key] : [];
	});
};
var extractUinFromCookie = (cookie) => {
	return extractCookieValue(cookie, "uin");
};
var setRequestCookieContext = (ctx, cookie) => {
	if (!cookie) return;
	ctx.request.cookie = cookie;
	const state = ctx.state || (ctx.state = {});
	state.requestCookie = cookie;
};
var resolveRequestCookie = (ctx, options = {}) => {
	const fallbackMode = options.fallbackMode ?? service_config_default.fallbackMode;
	const useGlobalCookie = options.useGlobalCookie ?? false;
	const cookieParamName = options.cookieParamName ?? service_config_default.cookieParamName;
	const query = ctx.query || {};
	const headers = ctx.headers || {};
	const request = ctx.request || {};
	if (fallbackMode) {
		const queryCookie = normalizeCookieValue(query[cookieParamName]);
		if (queryCookie) return {
			cookie: queryCookie,
			source: "query"
		};
		const customHeaderCookie = normalizeCookieValue(headers["x-custom-cookie"]);
		if (customHeaderCookie) return {
			cookie: customHeaderCookie,
			source: "x-custom-cookie"
		};
		const headerCookie = normalizeCookieValue(headers.cookie);
		if (headerCookie) return {
			cookie: headerCookie,
			source: "header-cookie"
		};
	}
	const requestCookie = normalizeCookieValue(request.cookie);
	if (requestCookie) return {
		cookie: requestCookie,
		source: "request"
	};
	if (useGlobalCookie) {
		const globalCookie = normalizeCookieValue(getUserCookie());
		if (globalCookie) return {
			cookie: globalCookie,
			source: "global"
		};
	}
	return { source: "none" };
};
//#endregion
export { setRequestCookieContext as a, getUserUin as c, getConfigDir as d, resolveConfigPath as f, resolveRequestCookie as i, initializeUserInfo as l, extractUinFromCookie as n, service_config_default as o, logger as p, getCookieKeys as r, getUserInfo as s, extractCookieValue as t, setUserInfo as u };
