// @sansenjian/qq-music-api/dist/cookieResolver-In4BnxUe.cjs
//#region \0rolldown/runtime.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __exportAll = (all, no_symbols) => {
	let target = {};
	for (var name in all) __defProp(target, name, {
		get: all[name],
		enumerable: true
	});
	if (!no_symbols) __defProp(target, Symbol.toStringTag, { value: "Module" });
	return target;
};
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));
//#endregion
let node_fs = require("node:fs");
node_fs = __toESM(node_fs, 1);
let node_path = require("node:path");
node_path = __toESM(node_path, 1);
let node_url = require("node:url");
//#region src/util/logger.ts
var SENSITIVE_KEY_PATTERN = /(cookie|authorization|token|uin|qq|user)/i;
var shouldLog = (level) => {
	if (process.env.NODE_ENV === "test") return false;
	if (level === "debug") return process.env.DEBUG === "true";
	return true;
};
var maskSensitiveValue = (value) => {
	if (typeof value === "string") return value ? "[masked]" : value;
	if (Array.isArray(value)) return value.map(maskSensitiveValue);
	if (value && typeof value === "object") return sanitizeLogPayload(value);
	return value;
};
var sanitizeLogPayload = (payload) => {
	return Object.fromEntries(Object.entries(payload).map(([key, value]) => [key, SENSITIVE_KEY_PATTERN.test(key) ? maskSensitiveValue(value) : maskSensitiveValueIfNested(value)]));
};
var maskSensitiveValueIfNested = (value) => {
	if (Array.isArray(value)) return value.map(maskSensitiveValueIfNested);
	if (value && typeof value === "object") return sanitizeLogPayload(value);
	return value;
};
var writeLog = (level, message, payload) => {
	if (!shouldLog(level)) return;
	const args = payload ? [message, sanitizeLogPayload(payload)] : [message];
	if (level === "error") {
		console.error(...args);
		return;
	}
	if (level === "warn") {
		console.warn(...args);
		return;
	}
	console.log(...args);
};
var logger = {
	debug: (message, payload) => writeLog("debug", message, payload),
	info: (message, payload) => writeLog("info", message, payload),
	warn: (message, payload) => writeLog("warn", message, payload),
	error: (message, payload) => writeLog("error", message, payload)
};
//#endregion
//#region src/config/config-path.ts
var moduleDir = node_path.default.dirname((0, node_url.fileURLToPath)(require("url").pathToFileURL(__filename).href));
var configDirName = "config";
var distDirName = "dist";
var srcDirName = "src";
var resolvePackageRoot = (fromDir) => {
	const baseName = node_path.default.basename(fromDir);
	if (baseName === configDirName) {
		const parentDir = node_path.default.dirname(fromDir);
		if (node_path.default.basename(parentDir) === srcDirName) return node_path.default.dirname(parentDir);
		return node_path.default.basename(parentDir) === distDirName ? node_path.default.dirname(parentDir) : parentDir;
	}
	if (baseName === srcDirName) return node_path.default.dirname(fromDir);
	return baseName === distDirName ? node_path.default.dirname(fromDir) : fromDir;
};
var resolveDefaultConfigDir = (fromDir = moduleDir) => node_path.default.join(resolvePackageRoot(fromDir), configDirName);
var getConfigDir = () => {
	if (process.env.QQ_MUSIC_API_CONFIG_DIR) return node_path.default.resolve(process.env.QQ_MUSIC_API_CONFIG_DIR);
	return resolveDefaultConfigDir();
};
var resolveConfigPath = (filename) => node_path.default.join(getConfigDir(), filename);
//#endregion
//#region src/config/user-info.ts
var userInfo = {
	loginUin: "",
	cookie: "",
	cookieList: [],
	cookieObject: {},
	refreshData: () => ({})
};
var cookieList = [];
var cookieObject = {};
var infoPath = resolveConfigPath("user-info.json");
var ensureConfigDir = () => {
	node_fs.default.mkdirSync(node_path.default.dirname(infoPath), { recursive: true });
};
var parseCookieObject = (cookie) => {
	const parsed = {};
	cookie.split(";").map((item) => item.trim()).filter(Boolean).forEach((item) => {
		const separatorIndex = item.indexOf("=");
		if (separatorIndex <= 0) return;
		const key = item.slice(0, separatorIndex).trim();
		const value = item.slice(separatorIndex + 1).trim();
		if (key) parsed[key] = value;
	});
	return parsed;
};
var normalizeUserInfo = (value) => {
	const parsed = value && typeof value === "object" ? value : {};
	return {
		loginUin: typeof parsed.loginUin === "string" ? parsed.loginUin : "",
		cookie: typeof parsed.cookie === "string" ? parsed.cookie : "",
		cookieList: [],
		cookieObject: {},
		refreshData: () => ({})
	};
};
var initData = () => {
	try {
		userInfo = normalizeUserInfo(JSON.parse(node_fs.default.readFileSync(infoPath, "utf-8")));
	} catch {
		userInfo = {
			loginUin: "",
			cookie: "",
			cookieList: [],
			cookieObject: {},
			refreshData: () => ({})
		};
	}
	cookieList = (userInfo.cookie || "").split(";").map((item) => item.trim()).filter(Boolean);
	cookieObject = parseCookieObject(userInfo.cookie || "");
};
var refreshData = (cookie) => {
	const uinMatch = cookie.match(/(?:^|;\s*)uin=([^;]+)/);
	const uin = uinMatch ? uinMatch[1] : "";
	ensureConfigDir();
	node_fs.default.writeFileSync(infoPath, JSON.stringify({
		loginUin: uin,
		cookie
	}), "utf-8");
	initData();
	return {
		...userInfo,
		uin: userInfo.loginUin || cookieObject.uin,
		cookieList,
		cookieObject
	};
};
initData();
var exportObj = {
	loginUin: userInfo?.loginUin || "",
	uin: userInfo?.loginUin || cookieObject?.uin || "",
	cookie: userInfo?.cookie || "",
	cookieList: cookieList || [],
	cookieObject: cookieObject || {},
	refreshData
};
//#endregion
//#region src/config/user-info-store.ts
var createEmptyUserInfo = () => ({
	loginUin: "",
	uin: "",
	cookie: "",
	cookieList: [],
	cookieObject: {},
	refreshData: () => ({})
});
var setUserInfo = (userInfo) => {
	global.userInfo = userInfo || createEmptyUserInfo();
	return global.userInfo;
};
var initializeUserInfo = (userInfo = exportObj) => setUserInfo(userInfo);
var getUserInfo = () => {
	if (!global.userInfo) return initializeUserInfo();
	return global.userInfo;
};
var getUserCookie = () => getUserInfo().cookie || "";
var getUserUin = (fallback = "0") => {
	const userInfo = getUserInfo();
	return String(userInfo.uin || userInfo.loginUin || fallback);
};
//#endregion
//#region src/config/service-config.ts
var configPath = resolveConfigPath("service-config.json");
var defaultConfig = {
	fallbackMode: true,
	useGlobalCookie: false,
	cookieParamName: "cookie"
};
var isRecord = (value) => Boolean(value && typeof value === "object");
var normalizeServiceConfig = (value) => {
	if (!isRecord(value)) return defaultConfig;
	return {
		fallbackMode: typeof value.fallbackMode === "boolean" ? value.fallbackMode : defaultConfig.fallbackMode,
		useGlobalCookie: typeof value.useGlobalCookie === "boolean" ? value.useGlobalCookie : defaultConfig.useGlobalCookie,
		cookieParamName: typeof value.cookieParamName === "string" && value.cookieParamName.trim() ? value.cookieParamName.trim() : defaultConfig.cookieParamName
	};
};
var config = defaultConfig;
if (node_fs.default.existsSync(configPath)) try {
	const content = node_fs.default.readFileSync(configPath, "utf-8");
	config = normalizeServiceConfig(JSON.parse(content));
} catch {
	logger.warn("service-config.json 读取失败，使用默认配置");
}
if (process.env.FALLBACK_MODE === "true") config.fallbackMode = true;
if (process.env.USE_GLOBAL_COOKIE === "true") config.useGlobalCookie = true;
var service_config_default = config;
//#endregion
//#region src/util/cookieResolver.ts
var normalizeCookieValue = (value) => {
	if (Array.isArray(value)) return normalizeCookieValue(value[0]);
	if (typeof value !== "string") return;
	const normalized = value.trim();
	return normalized ? normalized : void 0;
};
var extractCookieValue = (cookie, name) => {
	if (!cookie) return void 0;
	const escapedName = name.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
	return cookie.match(new RegExp(`(?:^|;\\s*)${escapedName}=([^;]*)`))?.[1]?.trim() || void 0;
};
var getCookieKeys = (cookie) => {
	if (!cookie) return [];
	return cookie.split(";").map((item) => item.trim()).filter(Boolean).flatMap((item) => {
		const separatorIndex = item.indexOf("=");
		if (separatorIndex <= 0) return [];
		const key = item.slice(0, separatorIndex).trim();
		return key ? [key] : [];
	});
};
var extractUinFromCookie = (cookie) => {
	return extractCookieValue(cookie, "uin");
};
var setRequestCookieContext = (ctx, cookie) => {
	if (!cookie) return;
	ctx.request.cookie = cookie;
	const state = ctx.state || (ctx.state = {});
	state.requestCookie = cookie;
};
var resolveRequestCookie = (ctx, options = {}) => {
	const fallbackMode = options.fallbackMode ?? service_config_default.fallbackMode;
	const useGlobalCookie = options.useGlobalCookie ?? false;
	const cookieParamName = options.cookieParamName ?? service_config_default.cookieParamName;
	const query = ctx.query || {};
	const headers = ctx.headers || {};
	const request = ctx.request || {};
	if (fallbackMode) {
		const queryCookie = normalizeCookieValue(query[cookieParamName]);
		if (queryCookie) return {
			cookie: queryCookie,
			source: "query"
		};
		const customHeaderCookie = normalizeCookieValue(headers["x-custom-cookie"]);
		if (customHeaderCookie) return {
			cookie: customHeaderCookie,
			source: "x-custom-cookie"
		};
		const headerCookie = normalizeCookieValue(headers.cookie);
		if (headerCookie) return {
			cookie: headerCookie,
			source: "header-cookie"
		};
	}
	const requestCookie = normalizeCookieValue(request.cookie);
	if (requestCookie) return {
		cookie: requestCookie,
		source: "request"
	};
	if (useGlobalCookie) {
		const globalCookie = normalizeCookieValue(getUserCookie());
		if (globalCookie) return {
			cookie: globalCookie,
			source: "global"
		};
	}
	return { source: "none" };
};
//#endregion
Object.defineProperty(exports, "__exportAll", {
	enumerable: true,
	get: function() {
		return __exportAll;
	}
});
Object.defineProperty(exports, "__toESM", {
	enumerable: true,
	get: function() {
		return __toESM;
	}
});
Object.defineProperty(exports, "extractCookieValue", {
	enumerable: true,
	get: function() {
		return extractCookieValue;
	}
});
Object.defineProperty(exports, "extractUinFromCookie", {
	enumerable: true,
	get: function() {
		return extractUinFromCookie;
	}
});
Object.defineProperty(exports, "getConfigDir", {
	enumerable: true,
	get: function() {
		return getConfigDir;
	}
});
Object.defineProperty(exports, "getCookieKeys", {
	enumerable: true,
	get: function() {
		return getCookieKeys;
	}
});
Object.defineProperty(exports, "getUserInfo", {
	enumerable: true,
	get: function() {
		return getUserInfo;
	}
});
Object.defineProperty(exports, "getUserUin", {
	enumerable: true,
	get: function() {
		return getUserUin;
	}
});
Object.defineProperty(exports, "initializeUserInfo", {
	enumerable: true,
	get: function() {
		return initializeUserInfo;
	}
});
Object.defineProperty(exports, "logger", {
	enumerable: true,
	get: function() {
		return logger;
	}
});
Object.defineProperty(exports, "resolveConfigPath", {
	enumerable: true,
	get: function() {
		return resolveConfigPath;
	}
});
Object.defineProperty(exports, "resolveRequestCookie", {
	enumerable: true,
	get: function() {
		return resolveRequestCookie;
	}
});
Object.defineProperty(exports, "service_config_default", {
	enumerable: true,
	get: function() {
		return service_config_default;
	}
});
Object.defineProperty(exports, "setRequestCookieContext", {
	enumerable: true,
	get: function() {
		return setRequestCookieContext;
	}
});
Object.defineProperty(exports, "setUserInfo", {
	enumerable: true,
	get: function() {
		return setUserInfo;
	}
});
