// @sansenjian/qq-music-api/dist/sdk.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_services = require("./services.cjs");
//#region src/sdk.ts
var withCookie = (option = {}, cookie) => {
	if (!cookie) return option;
	const headers = {
		...option.headers,
		Cookie: cookie
	};
	return {
		...option,
		headers
	};
};
var search = async ({ key, limit = 10, page = 1, catZhida = 1, remoteplace = "song", option = {} }) => require_services.getSearchByKey_default({
	method: "get",
	params: {
		w: key,
		n: limit,
		p: page,
		catZhida,
		remoteplace: `txt.yqq.${remoteplace}`
	},
	option
});
var getPlayUrl = async ({ songmid, quality, resType = "play", mediaId, cookie, option = {} }) => require_services.getMusicPlay_default({
	method: "get",
	params: {
		songmid,
		quality,
		resType,
		mediaId
	},
	option: withCookie(option, cookie)
});
var lyric = async ({ songmid, songid, isFormat = false, cookie, option = {} }) => require_services.getLyric_default({
	method: "get",
	params: {
		songmid,
		songid
	},
	isFormat,
	option: withCookie(option, cookie)
});
var getLoginQr = async () => require_services.getQQLoginQr$1({});
var checkLoginQr = async ({ ptqrtoken, qrsig }) => require_services.checkQQLoginQr$1({
	method: "post",
	params: {
		ptqrtoken,
		qrsig
	}
});
//#endregion
exports.checkLoginQr = checkLoginQr;
exports.checkQQLoginQr = checkLoginQr;
exports.getLoginQr = getLoginQr;
exports.getQQLoginQr = getLoginQr;
exports.getLyric = lyric;
exports.lyric = lyric;
exports.getMusicPlay = getPlayUrl;
exports.getPlayUrl = getPlayUrl;
exports.search = search;
