// @sansenjian/qq-music-api/dist/sdk.js
import { checkQQLoginQr, getLyric as getLyric_default, getMusicPlay as getMusicPlay_default, getQQLoginQr, getSearchByKey as getSearchByKey_default } from "./services.js";
//#region src/sdk.ts
var withCookie = (option = {}, cookie) => {
	if (!cookie) return option;
	const headers = {
		...option.headers,
		Cookie: cookie
	};
	return {
		...option,
		headers
	};
};
var search = async ({ key, limit = 10, page = 1, catZhida = 1, remoteplace = "song", option = {} }) => getSearchByKey_default({
	method: "get",
	params: {
		w: key,
		n: limit,
		p: page,
		catZhida,
		remoteplace: `txt.yqq.${remoteplace}`
	},
	option
});
var getPlayUrl = async ({ songmid, quality, resType = "play", mediaId, cookie, option = {} }) => getMusicPlay_default({
	method: "get",
	params: {
		songmid,
		quality,
		resType,
		mediaId
	},
	option: withCookie(option, cookie)
});
var lyric = async ({ songmid, songid, isFormat = false, cookie, option = {} }) => getLyric_default({
	method: "get",
	params: {
		songmid,
		songid
	},
	isFormat,
	option: withCookie(option, cookie)
});
var getLoginQr = async () => getQQLoginQr({});
var checkLoginQr = async ({ ptqrtoken, qrsig }) => checkQQLoginQr({
	method: "post",
	params: {
		ptqrtoken,
		qrsig
	}
});
//#endregion
export { checkLoginQr, checkLoginQr as checkQQLoginQr, getLoginQr, getLoginQr as getQQLoginQr, lyric as getLyric, lyric, getPlayUrl as getMusicPlay, getPlayUrl, search };
