// @sansenjian/qq-music-api/dist/services.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_cookieResolver = require("./cookieResolver-In4BnxUe.cjs");
let axios = require("axios");
axios = require_cookieResolver.__toESM(axios, 1);
let http = require("http");
http = require_cookieResolver.__toESM(http, 1);
let https = require("https");
https = require_cookieResolver.__toESM(https, 1);
let node_util = require("node:util");
//#region src/util/observability.ts
var now = () => Date.now();
var startTimer = () => {
	const startedAt = now();
	return () => now() - startedAt;
};
var logControllerStart = (name, ctx) => {
	require_cookieResolver.logger.debug(`[controller:${name}] start`, {
		method: ctx.method,
		path: ctx.path,
		query: ctx.query,
		params: ctx.params
	});
};
var logControllerSuccess = (name, ctx, durationMs) => {
	require_cookieResolver.logger.info(`[controller:${name}] success`, {
		status: ctx.status,
		durationMs
	});
};
var logControllerFailure = (name, error, durationMs) => {
	require_cookieResolver.logger.error(`[controller:${name}] failure`, {
		durationMs,
		error: error instanceof Error ? {
			name: error.name,
			message: error.message
		} : error
	});
};
var logServiceRequest = (name, payload) => {
	require_cookieResolver.logger.debug(`[service:${name}] request`, payload);
};
var logServiceSuccess = (name, response, durationMs) => {
	require_cookieResolver.logger.debug(`[service:${name}] success`, {
		status: response.status,
		durationMs
	});
};
var logServiceFailure = (name, error, durationMs) => {
	require_cookieResolver.logger.warn(`[service:${name}] failure`, {
		durationMs,
		error: error instanceof Error ? {
			name: error.name,
			message: error.message
		} : error
	});
};
var observeService = async (name, payload, callback) => {
	const duration = startTimer();
	logServiceRequest(name, payload);
	try {
		const response = await callback();
		logServiceSuccess(name, response, duration());
		return response;
	} catch (error) {
		logServiceFailure(name, error, duration());
		throw error;
	}
};
//#endregion
//#region src/util/colors.ts
var colors = {
	silly: (text) => (0, node_util.styleText)("magenta", text),
	input: (text) => (0, node_util.styleText)("gray", text),
	verbose: (text) => (0, node_util.styleText)("cyan", text),
	prompt: (text) => (0, node_util.styleText)("white", text),
	info: (text) => (0, node_util.styleText)("green", text),
	data: (text) => (0, node_util.styleText)("gray", text),
	help: (text) => (0, node_util.styleText)("cyan", text),
	warn: (text) => (0, node_util.styleText)("yellow", text),
	debug: (text) => (0, node_util.styleText)("blue", text),
	error: (text) => (0, node_util.styleText)("red", text)
};
//#endregion
//#region src/util/request.ts
var service = axios.default.create({
	withCredentials: true,
	timeout: 15e3,
	responseType: "json",
	httpAgent: new http.default.Agent({ keepAlive: true }),
	httpsAgent: new https.default.Agent({ keepAlive: true })
});
var ensureContentType = (config) => {
	const method = (config.method || "get").toLowerCase();
	const hasBody = config.data !== void 0 && config.data !== null;
	const headers = config.headers || {};
	const hasContentType = Boolean(headers["Content-Type"] || headers["content-type"]);
	if (hasBody && !hasContentType && [
		"post",
		"put",
		"patch",
		"delete"
	].includes(method)) headers["Content-Type"] = "application/x-www-form-urlencoded; charset=UTF-8";
	config.headers = headers;
};
service.interceptors.request.use((config) => {
	if (config.headers && !config.headers["User-Agent"]) config.headers["User-Agent"] = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36";
	ensureContentType(config);
	return config;
}, (error) => Promise.reject(error));
service.interceptors.response.use((response) => {
	if (!response) throw Error("response is null");
	if (process.env.DEBUG === "true") console.log(colors.info(`${response.config.url} request success`));
	return response;
}, (error) => {
	const url = error.config ? error.config.url : "Unknown URL";
	console.log(colors.error(`${url} request error: ${error.message}`));
	return Promise.reject(error);
});
var yURL = "https://y.qq.com";
var cURL = "https://c.y.qq.com";
var BASE_URL_MAP = {
	y: yURL,
	c: cURL
};
function request(config) {
	const url = config.url || "";
	const reqMethod = (config.method || "GET").toLowerCase();
	const baseURL = config.isUUrl === "u" ? url : (BASE_URL_MAP[config.isUUrl || "c"] || cURL) + url;
	const axiosConfig = {
		...config.options,
		url: baseURL,
		method: reqMethod
	};
	const headers = axiosConfig.headers || {};
	if (headers.cookies) {
		if (!headers.Cookie) headers.Cookie = headers.cookies;
		delete headers.cookies;
	}
	if (!headers.Cookie && headers.cookie) {
		headers.Cookie = headers.cookie;
		delete headers.cookie;
	}
	if (!headers.Cookie && config.cookie) headers.Cookie = config.cookie;
	axiosConfig.headers = headers;
	return service(axiosConfig);
}
//#endregion
//#region src/services/apis/downloadQQMusic.ts
var downloadQQMusic_default = ({ method = "get", params = {}, option = {} }) => {
	const data = {
		...params,
		format: "jsonp",
		jsonpCallback: "MusicJsonCallback",
		platform: "yqq"
	};
	return request({
		url: "/download/download.js",
		method,
		options: {
			...option,
			headers: {
				host: "y.qq.com",
				referer: "https://y.qq.com/",
				...option.headers
			},
			params: data
		},
		isUUrl: "y"
	}).then((res) => {
		let response = res.data;
		if (typeof response === "string") {
			const matches = response.match(/^\w+\(({[^()]+})\)$/);
			if (matches) response = JSON.parse(matches[1]);
		}
		return {
			status: 200,
			body: { response }
		};
	}).catch((error) => {
		console.log("error", error);
		return {
			status: 502,
			body: { error: error instanceof Error ? error.message : error }
		};
	});
};
//#endregion
//#region src/services/config.ts
var getCommonParams = () => ({
	g_tk: 1124214810,
	loginUin: require_cookieResolver.getUserUin(),
	hostUin: 0,
	inCharset: "utf8",
	outCharset: "utf-8",
	notice: 0,
	platform: "yqq.json",
	needNewCode: 0
});
var _guid = Math.round(2147483647 * Math.random()) * (/* @__PURE__ */ new Date()).getUTCMilliseconds() % 1e10;
//#endregion
//#region src/services/apis/y_common.ts
var y_common_default = ({ url, method = "get", options = {}, hasCommonParams = true }) => {
	const opts = { ...options };
	if (hasCommonParams) opts.params = {
		...getCommonParams(),
		...opts.params
	};
	else opts.params = { ...opts.params };
	opts.headers = {
		referer: "https://c.y.qq.com/",
		host: "c.y.qq.com",
		...opts.headers
	};
	if (process.env.DEBUG === "true") {
		const logOpts = {
			...opts,
			headers: { ...opts.headers }
		};
		const SENSITIVE_HEADER_KEYS = [
			"cookie",
			"authorization",
			"proxy-authorization"
		];
		if (logOpts.headers) Object.keys(logOpts.headers).forEach((key) => {
			if (SENSITIVE_HEADER_KEYS.includes(key.toLowerCase())) logOpts.headers[key] = "[masked]";
		});
		console.log(url, { opts: logOpts });
	}
	return request({
		url,
		method,
		options: opts
	});
};
//#endregion
//#region src/util/apiResponse.ts
var INTERNAL_ERROR_MESSAGE = "服务器内部错误";
/**
* 创建错误的 API 响应
* @param error 错误信息
* @param status HTTP 状态码，默认 500
*/
function errorResponse(error, status = 500) {
	return {
		status,
		body: { error: status >= 500 && error instanceof Error ? INTERNAL_ERROR_MESSAGE : error }
	};
}
/**
* 处理 API Promise，自动包装成功和错误响应
* @param promise API 请求 Promise
* @param options 可选配置
*/
async function handleApi(promise, options) {
	try {
		const result = await promise;
		const resultAny = result;
		const rawData = resultAny && typeof resultAny === "object" && Object.prototype.hasOwnProperty.call(resultAny, "data") ? resultAny.data : result;
		const responseData = options?.transformData ? options.transformData(rawData) : rawData;
		return {
			status: options?.customStatus || 200,
			body: { response: responseData }
		};
	} catch (error) {
		if (options?.logError !== false && process.env.NODE_ENV !== "test") console.log("API error:", error);
		return {
			status: options?.customStatus || 500,
			body: { error: INTERNAL_ERROR_MESSAGE }
		};
	}
}
/**
* 创建自定义结构的响应
* @param body 响应体
* @param status HTTP 状态码
*/
function customResponse(body, status = 200) {
	return {
		status,
		body
	};
}
//#endregion
//#region src/services/apis/search/getHotKey.ts
var getHotKey_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = {
		...params,
		format: "json",
		outCharset: "utf-8",
		hostUin: 0,
		needNewCode: 0
	};
	return handleApi(y_common_default({
		url: "/splcloud/fcgi-bin/gethotkey.fcg",
		method,
		options: {
			...option,
			params: data
		}
	}));
};
//#endregion
//#region src/services/apis/search/getSearchByKey.ts
var getSearchByKey_default = async ({ method = "get", params = {}, option = {} }) => {
	const { key, remoteplace, ...restParams } = params;
	const data = {
		format: "json",
		outCharset: "utf-8",
		ct: 24,
		qqmusic_ver: 1298,
		remoteplace: typeof remoteplace === "string" && remoteplace.trim() ? remoteplace.trim() : "txt.yqq.song",
		t: 0,
		aggr: 1,
		cr: 1,
		lossless: 0,
		flag_qc: 0,
		platform: "yqq.json",
		...restParams,
		...restParams.w === void 0 && key !== void 0 ? { w: key } : {}
	};
	const options = {
		...option,
		params: data
	};
	return observeService("getSearchByKey", {
		method,
		params: data
	}, () => handleApi(y_common_default({
		url: "/soso/fcgi-bin/client_search_cp",
		method,
		options
	})));
};
//#endregion
//#region src/services/apis/search/getSmartbox.ts
var getSmartbox_default = ({ method = "get", params = {}, option = {} }) => {
	const data = {
		...params,
		format: "json",
		outCharset: "utf-8",
		is_xml: 0
	};
	return y_common_default({
		url: "/splcloud/fcgi-bin/smartbox_new.fcg",
		method,
		options: {
			...option,
			params: data
		}
	}).then((res) => {
		return {
			status: 200,
			body: { response: res.data }
		};
	}).catch((error) => {
		console.log("error", error);
		return {
			status: 500,
			body: { error }
		};
	});
};
//#endregion
//#region src/services/apis/songLists/songLists.ts
var songLists_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = {
		...params,
		format: "json",
		outCharset: "utf-8",
		picmid: 1
	};
	return handleApi(y_common_default({
		url: "/splcloud/fcgi-bin/fcg_get_diss_by_tag.fcg",
		method,
		options: {
			...option,
			params: data
		}
	}), { transformData: (response) => {
		if (typeof response === "string") {
			const matches = response.match(/^\w+\(([^()]+)\)$/);
			if (matches) return JSON.parse(matches[1]);
		}
		return response;
	} });
};
//#endregion
//#region src/services/apis/songLists/songListCategories.ts
var songListCategories_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = {
		...params,
		format: "json",
		outCharset: "utf-8"
	};
	return handleApi(y_common_default({
		url: "/splcloud/fcgi-bin/fcg_get_diss_tag_conf.fcg",
		method,
		options: {
			...option,
			params: data
		}
	}));
};
//#endregion
//#region src/services/apis/songLists/songListDetail.ts
var songListDetail_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = {
		...params,
		format: "json",
		outCharset: "utf-8",
		type: 1,
		json: 1,
		utf8: 1,
		onlysong: 0,
		new_format: 1
	};
	return handleApi(y_common_default({
		url: "/qzone/fcg-bin/fcg_ucc_getcdinfo_byids_cp.fcg",
		method,
		options: {
			...option,
			params: data
		}
	}));
};
//#endregion
//#region src/services/apis/mv/getMvByTag.ts
var getMvByTag_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = Object.assign(params, {
		format: "json",
		outCharset: "GB2312",
		cmd: "shoubo",
		lan: "all"
	});
	return handleApi(request({
		url: "/mv/fcgi-bin/getmv_by_tag",
		method,
		options: Object.assign(option, { params: data })
	}));
};
//#endregion
//#region src/services/apis/u_common.ts
var u_common_default = ({ options = {}, method = "get", customCookie }) => {
	const opts = { ...options };
	opts.params = {
		...getCommonParams(),
		...opts.params
	};
	opts.headers = {
		referer: "https://y.qq.com/portal/player.html",
		host: "u.y.qq.com",
		"content-type": "application/x-www-form-urlencoded",
		...opts.headers
	};
	if (process.env.DEBUG === "true") {
		const logOpts = {
			...opts,
			headers: {
				...opts.headers,
				cookie: "[REDACTED]"
			}
		};
		console.log("https://u.y.qq.com/cgi-bin/musicu.fcg", { opts: logOpts });
	}
	return request({
		url: "https://u.y.qq.com/cgi-bin/musicu.fcg",
		method,
		options: opts,
		isUUrl: "u",
		cookie: customCookie
	});
};
//#endregion
//#region src/services/apis/musicu.ts
var callMusicu = (data, { method = "POST", option = {}, cookie } = {}) => {
	const optionRecord = option;
	const headers = {
		...optionRecord.headers,
		"Content-Type": optionRecord.headers?.["Content-Type"] || "application/json"
	};
	return handleApi(u_common_default({
		method,
		customCookie: cookie,
		options: {
			...optionRecord,
			data: JSON.stringify(data),
			headers
		}
	}));
};
var callMusicuQuery = (data, { method = "GET", option = {}, cookie } = {}) => {
	const optionRecord = option;
	return handleApi(u_common_default({
		method,
		customCookie: cookie,
		options: {
			...optionRecord,
			params: {
				...optionRecord.params,
				format: "json",
				data: JSON.stringify(data)
			}
		}
	}));
};
//#endregion
//#region src/services/apis/mv/getMvCategory.ts
var getMvCategory_default = async ({ method = "POST", option = {} }) => {
	return callMusicuQuery({
		comm: {
			...getCommonParams(),
			ct: 24,
			cv: 0
		},
		mv_tag: {
			module: "MvService.MvInfoProServer",
			method: "GetAllocTag",
			param: {}
		}
	}, {
		method,
		option
	});
};
//#endregion
//#region src/services/apis/singers/getSingerCategory.ts
var getSingerCategory_default = async ({ method = "POST", option = {} }) => {
	return callMusicuQuery({
		comm: {
			...getCommonParams(),
			ct: 24,
			cv: 0
		},
		singerList: {
			module: "Music.SingerListServer",
			method: "get_singer_list",
			param: {
				area: -100,
				sex: -100,
				genre: -100,
				index: -100,
				sin: 0,
				cur_page: 1
			}
		}
	}, {
		method,
		option
	});
};
//#endregion
//#region src/services/apis/singers/getSimilarSinger.ts
var getSimilarSinger_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = Object.assign(params, {
		format: "json",
		outCharset: "utf-8",
		utf8: 1,
		start: 0,
		num: 5
	});
	return handleApi(request({
		url: "/v8/fcg-bin/fcg_v8_simsinger.fcg",
		method,
		options: Object.assign(option, { params: data })
	}));
};
//#endregion
//#region src/services/apis/singers/getSingerMv.ts
var getSingerMv_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = Object.assign(params, {
		format: "json",
		outCharset: "utf-8",
		cid: 205360581,
		begin: 0
	});
	return handleApi(request({
		url: "/mv/fcgi-bin/fcg_singer_mv.fcg",
		method,
		options: Object.assign(option, { params: data })
	}));
};
//#endregion
//#region src/services/apis/singers/getSingerDesc.ts
var getSingerDesc_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = Object.assign(params, {
		format: "xml",
		outCharset: "utf-8",
		utf8: 1,
		r: Date.now()
	});
	return handleApi(request({
		url: "/splcloud/fcgi-bin/fcg_get_singer_desc.fcg",
		method,
		options: Object.assign(option, { params: data }),
		isUUrl: "c"
	}));
};
//#endregion
//#region src/services/apis/singers/getSingerStarNum.ts
var getSingerStarNum_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = Object.assign(params, {
		format: "json",
		outCharset: "utf-8",
		utf8: 1,
		rnd: Date.now()
	});
	return handleApi(request({
		url: "/rsc/fcgi-bin/fcg_order_singer_getnum.fcg",
		method,
		options: Object.assign(option, { params: data })
	}));
};
//#endregion
//#region src/services/apis/radio/getRadioLists.ts
var getRadioLists_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = Object.assign(params, {
		format: "json",
		outCharset: "utf-8",
		channel: "radio",
		page: "index",
		tpl: "wk",
		new: 1,
		p: Math.round(1)
	});
	return handleApi(request({
		url: "/v8/fcg-bin/fcg_v8_radiolist.fcg",
		method,
		options: Object.assign(option, { params: data })
	}));
};
//#endregion
//#region src/services/apis/digitalAlbum/getDigitalAlbumLists.ts
var getDigitalAlbumLists_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = Object.assign(params, {
		format: "json",
		outCharset: "utf-8",
		cmd: "pc_index_new"
	});
	return handleApi(request({
		url: "/v8/fcg-bin/musicmall.fcg",
		method,
		options: Object.assign(option, { params: data })
	}));
};
//#endregion
//#region src/util/lyricParse.ts
var timeExp = /\[(\d{2,}):(\d{2})(?:\.(\d{2,3}))?]/g;
var TAGREGMAP = {
	title: "ti",
	artist: "ar",
	album: "al",
	offset: "offset",
	by: "by"
};
var Lyric = class {
	constructor(lyric) {
		this.lyric = lyric;
		this.tags = {};
		this.lines = [];
		this._init();
	}
	_init() {
		this._initTag();
		this._initLines();
	}
	_initTag() {
		for (const tag in TAGREGMAP) {
			const matches = this.lyric.match(new RegExp(`\\[${TAGREGMAP[tag]}:([^\\]]*)]`, "i"));
			this.tags[tag] = matches && matches[1] || "";
		}
	}
	_initLines() {
		const lines = this.lyric.split("\n");
		for (let i = 0; i < lines.length; i++) {
			const line = lines[i].trim();
			timeExp.lastIndex = 0;
			const matches = [...line.matchAll(timeExp)];
			if (matches.length > 0) {
				const txt = line.replace(timeExp, "").trim();
				if (txt) for (const result of matches) {
					const minutes = parseInt(result[1], 10);
					const seconds = parseInt(result[2], 10);
					const milliseconds = result[3] ? parseInt(result[3], 10) : 0;
					const time = minutes * 60 * 1e3 + seconds * 1e3 + milliseconds * 10;
					this.lines.push({
						time,
						txt
					});
				}
			}
		}
		this.lines.sort((a, b) => {
			return a.time - b.time;
		});
	}
};
function lyricParse(lyricString) {
	return new Lyric(lyricString);
}
//#endregion
//#region src/services/apis/UCommon/UCommon.ts
var UCommon_default = ({ method = "get", params = {}, option = {} }) => {
	return u_common_default({
		method,
		options: {
			...option,
			params
		}
	});
};
//#endregion
//#region src/services/apis/music/getLyric.ts
var MUSICU_TIMEOUT_MS = 1e4;
var getCookieFromOptions$1 = (option) => {
	if (!option || typeof option !== "object") return;
	const headers = option.headers;
	if (!headers || typeof headers !== "object") return;
	const cookie = headers.Cookie || headers.cookie;
	return typeof cookie === "string" ? cookie : void 0;
};
var resolveUin$1 = (cookie) => {
	return require_cookieResolver.extractUinFromCookie(cookie) || require_cookieResolver.getUserUin();
};
var decodeLyricField = (value) => {
	if (typeof value !== "string" || !value) return "";
	try {
		return Buffer.from(value, "base64").toString() || value;
	} catch {
		return value;
	}
};
var normalizeLyricResponse = (resData, isFormat) => {
	const lyricString = decodeLyricField(resData?.lyric);
	const lyric = isFormat && lyricString ? lyricParse(lyricString) : lyricString;
	return {
		...resData,
		lyric
	};
};
var hasNegativeBizCode = (data) => {
	return [
		data?.retcode,
		data?.code,
		data?.subcode
	].map((item) => Number(item)).filter((item) => !Number.isNaN(item)).some((item) => item < 0);
};
var normalizeSongId = (value) => {
	const id = Number(value);
	if (Number.isFinite(id) && id > 0) return String(id);
};
var mergePrimaryAndFallbackPayload = (primaryPayload, fallbackPayload) => {
	const merged = {
		...primaryPayload,
		...fallbackPayload
	};
	if (!Object.prototype.hasOwnProperty.call(fallbackPayload, "retcode")) merged.retcode = 0;
	if (!Object.prototype.hasOwnProperty.call(fallbackPayload, "code")) merged.code = 0;
	if (!Object.prototype.hasOwnProperty.call(fallbackPayload, "subcode")) merged.subcode = 0;
	return merged;
};
var resolveSongIdBySongmid = async ({ songmid, loginUin }) => {
	const data = (await UCommon_default({
		method: "get",
		params: {
			format: "json",
			data: {
				comm: {
					uin: loginUin,
					ct: 24,
					cv: 0
				},
				songinfo: {
					method: "get_song_detail_yqq",
					param: {
						song_type: 0,
						song_mid: songmid,
						song_id: 0
					},
					module: "music.pf_song_detail_svr"
				}
			}
		},
		option: {}
	}))?.data || {};
	return normalizeSongId(data?.songinfo?.data?.track_info?.id || data?.songinfo?.data?.trackInfo?.id || data?.track_info?.id);
};
var fetchLyricByMusicu = async ({ songmid, songid, loginUin, cookie }) => {
	const reqPayload = {
		req_0: {
			module: "music.musichallSong.PlayLyricInfo",
			method: "GetPlayLyricInfo",
			param: {
				songMID: songmid || "",
				songID: Number(songid || 0),
				trans_t: 0,
				roma_t: 0,
				qrc_t: 0,
				crypt: 1,
				lrc_t: 0,
				interval: 0
			}
		},
		loginUin,
		comm: {
			uin: loginUin,
			format: "json",
			ct: 24,
			cv: 0
		}
	};
	const controller = new AbortController();
	const timer = setTimeout(() => controller.abort(), MUSICU_TIMEOUT_MS);
	const headers = {
		"Content-Type": "application/json",
		Referer: "https://y.qq.com/portal/player.html",
		Origin: "https://y.qq.com"
	};
	if (cookie) headers.Cookie = cookie;
	const raw = await (await (async () => {
		try {
			return await fetch("https://u.y.qq.com/cgi-bin/musicu.fcg", {
				method: "POST",
				headers,
				body: JSON.stringify(reqPayload),
				signal: controller.signal
			});
		} finally {
			clearTimeout(timer);
		}
	})()).text();
	const upstream = JSON.parse(raw || "{}");
	return upstream?.req_0?.data || upstream?.PlayLyricInfo?.data || {};
};
var getLyric = async ({ method = "get", params = {}, option = {}, isFormat = false }) => {
	const cookie = getCookieFromOptions$1(option);
	const loginUin = resolveUin$1(cookie);
	const songmid = typeof params.songmid === "string" ? params.songmid : void 0;
	const songid = typeof params.songid === "string" ? params.songid : void 0;
	const data = Object.assign(params, {
		format: "json",
		outCharset: "utf-8",
		pcachetime: Date.now(),
		loginUin
	});
	const options = Object.assign(option, {
		headers: Object.assign({}, option?.headers || {}, {
			referer: "https://y.qq.com/portal/player.html",
			host: "c.y.qq.com"
		}),
		params: data
	});
	try {
		let payload = (await y_common_default({
			url: "/lyric/fcgi-bin/fcg_query_lyric_new.fcg",
			method,
			options
		}))?.data || {};
		if (hasNegativeBizCode(payload)) try {
			let fallbackPayload = await fetchLyricByMusicu({
				songmid,
				songid,
				loginUin,
				cookie
			});
			if (hasNegativeBizCode(fallbackPayload) && !normalizeSongId(songid) && songmid) {
				const resolvedSongId = await resolveSongIdBySongmid({
					songmid,
					loginUin
				});
				if (resolvedSongId) fallbackPayload = await fetchLyricByMusicu({
					songmid,
					songid: resolvedSongId,
					loginUin,
					cookie
				});
			}
			if (Object.keys(fallbackPayload).length > 0 && !hasNegativeBizCode(fallbackPayload)) payload = mergePrimaryAndFallbackPayload(payload, fallbackPayload);
		} catch {}
		return {
			status: 200,
			body: { response: normalizeLyricResponse(payload, Boolean(isFormat)) }
		};
	} catch (error) {
		return {
			status: 500,
			body: { error: error instanceof Error ? {
				name: error.name,
				message: error.message
			} : {
				name: "Error",
				message: "Internal server error"
			} }
		};
	}
};
var getLyric_default = (props) => observeService("getLyric", {
	method: props.method,
	params: props.params,
	option: props.option
}, () => getLyric(props));
//#endregion
//#region src/services/apis/music/getMusicPlay.ts
var DEFAULT_QUALITY = "128";
var FILE_TYPE_MAP = {
	m4a: {
		prefix: "C400",
		suffix: ".m4a"
	},
	128: {
		prefix: "M500",
		suffix: ".mp3"
	},
	320: {
		prefix: "M800",
		suffix: ".mp3"
	},
	ape: {
		prefix: "A000",
		suffix: ".ape"
	},
	flac: {
		prefix: "F000",
		suffix: ".flac"
	}
};
var normalizeFirstValue = (value) => {
	if (Array.isArray(value)) return normalizeFirstValue(value[0]);
	if (typeof value !== "string") return;
	return value.trim() || void 0;
};
var normalizeSongmidList = (songmid) => {
	return (Array.isArray(songmid) ? songmid : [songmid]).flatMap((value) => {
		if (typeof value !== "string") return [];
		return value.split(",");
	}).map((item) => item.trim()).filter(Boolean);
};
var normalizeQuality = (value) => {
	const quality = normalizeFirstValue(value) || String(value || "");
	if (quality === "m4a" || quality === "ape" || quality === "flac") return quality;
	const numericQuality = Number(quality);
	if (numericQuality === 320) return "320";
	if (numericQuality === 128) return "128";
	return DEFAULT_QUALITY;
};
var pickPlayableDomain = (sip) => {
	if (!Array.isArray(sip)) return "";
	const urls = sip.filter((item) => typeof item === "string" && item.length > 0);
	return urls.find((url) => !url.startsWith("http://ws")) || urls.find((url) => url.startsWith("https://")) || urls[0] || "";
};
var joinUrl = (domain, path) => {
	if (domain.endsWith("/") && path.startsWith("/")) return `${domain}${path.slice(1)}`;
	if (!domain.endsWith("/") && !path.startsWith("/")) return `${domain}/${path}`;
	return `${domain}${path}`;
};
var buildPlayUrl = (domain, info, guid) => {
	if (!domain) return "";
	if (info.purl) return joinUrl(domain, info.purl);
	if (info.vkey && info.filename) return `${joinUrl(domain, info.filename)}?vkey=${info.vkey}&guid=${guid}&fromtag=66`;
	return "";
};
var resolveUin = (cookie) => {
	return require_cookieResolver.extractUinFromCookie(cookie) || require_cookieResolver.getUserUin();
};
var getCookieFromOptions = (option) => {
	if (!option || typeof option !== "object") return;
	const headers = option.headers;
	if (!headers || typeof headers !== "object") return;
	const cookie = headers.Cookie || headers.cookie;
	return typeof cookie === "string" ? cookie : void 0;
};
var getMusicPlay_default = async ({ method = "get", params = {}, option = {} }) => {
	const musicParams = params;
	const songmidList = normalizeSongmidList(musicParams.songmid);
	if (songmidList.length === 0) return {
		status: 400,
		body: { data: { message: "no songmid" } }
	};
	const justPlayUrl = (normalizeFirstValue(musicParams.resType) || "play") === "play";
	const mediaId = normalizeFirstValue(musicParams.mediaId);
	const quality = normalizeQuality(musicParams.quality);
	const guid = String(_guid || "1429839143");
	const cookie = getCookieFromOptions(option);
	const uin = resolveUin(cookie);
	const authst = require_cookieResolver.extractCookieValue(cookie, "qqmusic_key");
	const fileType = FILE_TYPE_MAP[quality];
	const requestPayload = {
		req_0: {
			module: "vkey.GetVkeyServer",
			method: "CgiGetVkey",
			param: {
				filename: songmidList.map((item) => `${fileType.prefix}${item}${mediaId || item}${fileType.suffix}`),
				guid,
				songmid: songmidList,
				songtype: [0],
				uin,
				loginflag: 1,
				platform: "20",
				...authst ? { authst } : {}
			}
		},
		loginUin: uin,
		comm: {
			uin,
			format: "json",
			ct: 24,
			cv: 0
		}
	};
	const upstreamParams = {
		format: "json",
		data: JSON.stringify(requestPayload)
	};
	try {
		const upstreamData = (await UCommon_default({
			method,
			params: upstreamParams,
			option
		})).data;
		const domain = pickPlayableDomain(upstreamData?.req_0?.data?.sip);
		const midurlinfo = Array.isArray(upstreamData?.req_0?.data?.midurlinfo) ? upstreamData.req_0.data.midurlinfo : [];
		const playUrl = {};
		const midInfoMap = /* @__PURE__ */ new Map();
		midurlinfo.forEach((item) => {
			if (item.songmid) midInfoMap.set(item.songmid, item);
		});
		songmidList.forEach((songmid) => {
			const item = midInfoMap.get(songmid);
			const url = item ? buildPlayUrl(domain, item, guid) : "";
			playUrl[songmid] = {
				url,
				error: url ? void 0 : "暂无播放链接"
			};
		});
		upstreamData.playUrl = playUrl;
		return {
			status: 200,
			body: { data: justPlayUrl ? { playUrl } : upstreamData }
		};
	} catch (error) {
		return {
			status: 502,
			body: { error: error instanceof Error ? error.message : error }
		};
	}
};
//#endregion
//#region src/services/apis/music/getRelatedMv.ts
var getRelatedMv_default = async ({ method = "POST", params = {}, option = {} }) => {
	const songid = String(params.songid || params.id || "");
	return callMusicu({
		comm: {
			...getCommonParams(),
			format: "json",
			inCharset: "utf-8",
			outCharset: "utf-8",
			platform: "h5",
			needNewCode: 1
		},
		video: {
			module: "MvService.MvInfoProServer",
			method: "GetSongRelatedMv",
			param: {
				songid,
				songtype: Number(params.songtype || params.songType || 1),
				lastmvid: Number(params.lastmvid || params.lastMvId || 0),
				num: Number(params.num || params.limit || 10)
			}
		}
	}, {
		method,
		option
	});
};
//#endregion
//#region src/services/apis/music/getRelatedPlaylists.ts
var getRelatedPlaylists_default = async ({ method = "POST", params = {}, option = {} }) => {
	const songid = Number(params.songid || params.id || 0);
	return callMusicu({
		comm: {
			...getCommonParams(),
			format: "json",
			inCharset: "utf-8",
			outCharset: "utf-8",
			platform: "h5",
			needNewCode: 1
		},
		gedan: {
			module: "music.mb_gedan_recommend_svr",
			method: "get_related_gedan",
			param: {
				sin: Number(params.sin || 0),
				last_id: Number(params.lastId || params.last_id || 0),
				song_type: Number(params.songType || params.song_type || 1),
				song_id: songid
			}
		}
	}, {
		method,
		option
	});
};
//#endregion
//#region src/services/apis/album/getAlbumInfo.ts
var getAlbumInfo_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = {
		...params,
		format: "json",
		outCharset: "utf-8"
	};
	const options = {
		...option,
		params: data
	};
	return observeService("getAlbumInfo", {
		method,
		params: data
	}, () => handleApi(y_common_default({
		url: "/v8/fcg-bin/fcg_v8_album_info_cp.fcg",
		method,
		options
	})));
};
//#endregion
//#region src/services/apis/album/getAlbumSongs.ts
var getAlbumSongs_default = async ({ method = "POST", params = {}, option = {} }) => {
	const albummid = String(params.albummid || params.albumMid || "");
	return callMusicu({
		comm: {
			...getCommonParams(),
			ct: 24,
			cv: 1e4
		},
		albumSonglist: {
			method: "GetAlbumSongList",
			module: "music.musichallAlbum.AlbumSongList",
			param: {
				albumMid: albummid,
				albumID: Number(params.albumid || params.albumID || 0),
				begin: Number(params.begin ?? params.offset ?? 0),
				num: Number(params.num ?? params.limit ?? 999),
				order: Number(params.order ?? 2)
			}
		}
	}, {
		method,
		option
	});
};
//#endregion
//#region src/services/apis/comments/getComments.ts
var getComments_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = {
		...params,
		format: "json",
		outCharset: "GB2312",
		domain: "qq.com",
		ct: 24,
		cv: 10101010,
		needmusiccrit: 0
	};
	return handleApi(y_common_default({
		url: "/base/fcgi-bin/fcg_global_comment_h5.fcg",
		method,
		options: {
			...option,
			params: data
		}
	}));
};
//#endregion
//#region src/services/apis/rank/getTopLists.ts
var getTopLists_default = async ({ method = "get", params = {}, option = {} }) => {
	const data = Object.assign(params, {
		format: "json",
		outCharset: "utf-8",
		platform: "h5",
		needNewCode: 1
	});
	return handleApi(request({
		url: "/v8/fcg-bin/fcg_myqq_toplist.fcg",
		method,
		options: Object.assign(option, { params: data }),
		isUUrl: "c"
	}), { transformData: (response) => {
		if (typeof response === "string") {
			const matches = response.match(/^\w+\(({[^()]+})\)$/);
			if (matches) return JSON.parse(matches[1]);
		}
		return response;
	} });
};
//#endregion
//#region src/services/apis/recommend/getRecommendBanner.ts
var getRecommendBanner_default = async ({ method = "POST", option = {} }) => {
	return callMusicuQuery({
		comm: {
			...getCommonParams(),
			ct: 24,
			cv: 0
		},
		focus: {
			module: "QQMusic.MusichallServer",
			method: "GetFocus",
			param: {}
		}
	}, {
		method,
		option
	});
};
//#endregion
//#region src/util/loginUtils.ts
var hash33 = (t) => {
	let e = 0;
	for (let n = 0, o = t.length; n < o; ++n) e += (e << 5) + t.charCodeAt(n);
	return 2147483647 & e;
};
var getGtk = (p_skey) => {
	const str = p_skey;
	let hash = 5381;
	for (let i = 0, len = str.length; i < len; ++i) hash += (hash << 5) + str.charCodeAt(i);
	return hash & 2147483647;
};
var getGuid = () => {
	return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function(c) {
		const r = Math.random() * 16 | 0;
		return (c === "x" ? r : r & 3 | 8).toString(16);
	}).toUpperCase();
};
//#endregion
//#region src/services/apis/user/getQQLoginQr.ts
var getQQLoginQr = async (_props) => {
	try {
		const response = await fetch("https://ssl.ptlogin2.qq.com/ptqrshow?appid=716027609&e=2&l=M&s=3&d=72&v=4&t=0.9698127522807933&daid=383&pt_3rd_aid=100497308&u1=https%3A%2F%2Fgraph.qq.com%2Foauth2.0%2Flogin_jump");
		const data = await response.arrayBuffer();
		const img = "data:image/png;base64," + (data && Buffer.from(data).toString("base64"));
		const match = response.headers.get("Set-Cookie")?.match(/qrsig=([^;]+)/);
		if (!match) return errorResponse("Failed to get qrsig from response", 502);
		const qrsig = match[1];
		return customResponse({
			img,
			ptqrtoken: hash33(qrsig),
			qrsig
		}, 200);
	} catch {
		return errorResponse("Failed to fetch QQ login QR", 502);
	}
};
//#endregion
//#region src/services/apis/user/checkQQLoginQr.ts
var REQUEST_TIMEOUT_MS = 1e4;
var DEBUG_ENABLED$2 = process.env.DEBUG === "true";
var debugLog$2 = (message, payload) => {
	if (DEBUG_ENABLED$2) console.log(`[checkQQLoginQr] ${message}`, payload ?? "");
};
var parseSetCookie = (setCookieHeader) => {
	if (!setCookieHeader) return [];
	const cookies = [];
	const parts = setCookieHeader.split(/,(?=\s*[a-zA-Z_]+=)/);
	for (const part of parts) {
		const cookiePair = part.split(";")[0].trim();
		if (cookiePair && cookiePair.includes("=") && cookiePair.split("=")[1]) cookies.push(cookiePair);
	}
	return cookies;
};
var fetchWithTimeout = async (input, init = {}, timeout = REQUEST_TIMEOUT_MS) => {
	const controller = new AbortController();
	const timer = setTimeout(() => controller.abort(), timeout);
	try {
		return await fetch(input, {
			...init,
			signal: controller.signal
		});
	} finally {
		clearTimeout(timer);
	}
};
var buildLoginSession = (cookie) => {
	const cookieList = cookie.split(";").map((item) => item.trim()).filter(Boolean);
	const cookieObject = {};
	cookieList.forEach((item) => {
		const [key, value = ""] = item.split("=");
		if (key && value) cookieObject[key] = value;
	});
	const loginUin = cookieObject.uin || "";
	return {
		loginUin,
		uin: loginUin,
		cookie,
		cookieList,
		cookieObject
	};
};
var checkQQLoginQr = async ({ params = {} }) => {
	const { ptqrtoken, qrsig } = params;
	if (!ptqrtoken || !qrsig) return errorResponse("参数错误", 400);
	try {
		const response = await fetchWithTimeout(`https://ssl.ptlogin2.qq.com/ptqrlogin?u1=https%3A%2F%2Fgraph.qq.com%2Foauth2.0%2Flogin_jump&ptqrtoken=${ptqrtoken}&ptredirect=0&h=1&t=1&g=1&from_ui=1&ptlang=2052&action=0-0-1711022193435&js_ver=23111510&js_type=1&login_sig=du-YS1h8*0GqVqcrru0pXkpwVg2DYw-DtbFulJ62IgPf6vfiJe*4ONVrYc5hMUNE&pt_uistyle=40&aid=716027609&daid=383&pt_3rd_aid=100497308&&o1vId=3674fc47871e9c407d8838690b355408&pt_js_version=v1.48.1`, { headers: { Cookie: `qrsig=${qrsig}` } });
		const data = await response.text() || "";
		const cookieMap = /* @__PURE__ */ new Map();
		const setCookie = (setCookieHeader) => {
			const cookies = parseSetCookie(setCookieHeader);
			for (const cookie of cookies) {
				const [name] = cookie.split("=");
				cookieMap.set(name, cookie);
			}
		};
		setCookie(response.headers.get("Set-Cookie"));
		const refresh = data.includes("已失效");
		if (!data.includes("登录成功")) return customResponse({
			isOk: false,
			refresh,
			message: refresh && "二维码已失效" || "未扫描二维码"
		}, 200);
		const allCookie = () => Array.from(cookieMap.values());
		const urlMatch = data.match(/(?:'((?:https?|ftp):\/\/[^\s/$.?#].[^\s]*)')/g);
		if (!urlMatch || !urlMatch[0]) return errorResponse("登录检查失败", 502);
		const checkSigCookie = (await fetchWithTimeout(urlMatch[0].replace(/'/g, ""), {
			redirect: "manual",
			headers: { Cookie: allCookie().join("; ") }
		})).headers.get("Set-Cookie");
		const pSkeyMatch = checkSigCookie?.match(/p_skey=([^;]+)/);
		if (!pSkeyMatch || !pSkeyMatch[1]) return errorResponse("登录检查失败", 502);
		const p_skey = pSkeyMatch[1];
		const gtk = getGtk(p_skey);
		setCookie(checkSigCookie);
		const authorizeUrl = "https://graph.qq.com/oauth2.0/authorize";
		const getAuthorizeData = (g_tk) => {
			const data = new FormData();
			data.append("response_type", "code");
			data.append("client_id", "100497308");
			data.append("redirect_uri", "https://y.qq.com/portal/wx_redirect.html?login_type=1&surl=https://y.qq.com/");
			data.append("scope", "get_user_info,get_app_friends");
			data.append("state", "state");
			data.append("switch", "");
			data.append("from_ptlogin", "1");
			data.append("src", "1");
			data.append("update_auth", "1");
			data.append("openapi", "1010_1030");
			data.append("g_tk", g_tk.toString());
			data.append("auth_time", (/* @__PURE__ */ new Date()).toString());
			data.append("ui", getGuid());
			return data;
		};
		const authorizeRes = await fetchWithTimeout(authorizeUrl, {
			redirect: "manual",
			method: "POST",
			body: getAuthorizeData(gtk),
			headers: { Cookie: allCookie().join("; ") }
		});
		setCookie(authorizeRes.headers.get("Set-Cookie"));
		const location = authorizeRes.headers.get("Location");
		const contentType = authorizeRes.headers.get("Content-Type");
		debugLog$2("authorize response meta", {
			status: authorizeRes.status,
			redirected: authorizeRes.redirected,
			hasLocation: Boolean(location),
			contentType
		});
		if (authorizeRes.status < 300 || authorizeRes.status >= 400 || !location) return errorResponse("登录检查失败", 502);
		const codeMatch = location.match(/[?&]code=([^&]+)/);
		if (!codeMatch || !codeMatch[1]) {
			debugLog$2("authorize location parse failed", { location });
			return errorResponse("登录检查失败", 502);
		}
		const code = codeMatch[1];
		const getFcgReqData = (g_tk, code) => {
			return JSON.stringify({
				comm: {
					g_tk,
					platform: "yqq",
					ct: 24,
					cv: 0
				},
				req: {
					module: "QQConnectLogin.LoginServer",
					method: "QQLogin",
					param: { code }
				}
			});
		};
		setCookie((await fetchWithTimeout("https://u.y.qq.com/cgi-bin/musicu.fcg", {
			method: "POST",
			body: getFcgReqData(gtk, code),
			headers: {
				"Content-Type": "application/x-www-form-urlencoded",
				Cookie: allCookie().join("; ")
			}
		})).headers.get("Set-Cookie"));
		return customResponse({
			isOk: true,
			message: "登录成功",
			session: buildLoginSession(allCookie().join("; "))
		}, 200);
	} catch (error) {
		if (error.name === "AbortError") return errorResponse("登录检查超时", 504);
		return errorResponse("登录检查失败", 502);
	}
};
//#endregion
//#region src/services/apis/user/getUserPlaylists.ts
var DEBUG_ENABLED$1 = process.env.DEBUG === "true";
var debugLog$1 = (message, payload) => {
	if (DEBUG_ENABLED$1) console.log(`[getUserPlaylists] ${message}`, payload ?? "");
};
var getNamedCandidateEntries = (payload) => [
	["data.mydiss.list", payload?.data?.mydiss?.list],
	["data.mymusic", payload?.data?.mymusic],
	["data.createdDissList", payload?.data?.createdDissList],
	["data.createdList", payload?.data?.createdList],
	["data.creator", payload?.data?.creator],
	["data.creator.playlist", payload?.data?.creator?.playlist],
	["data.creator.playlists", payload?.data?.creator?.playlists],
	["data.playlist", payload?.data?.playlist],
	["data.playlists", payload?.data?.playlists],
	["mydiss.list", payload?.mydiss?.list],
	["mymusic", payload?.mymusic],
	["createdDissList", payload?.createdDissList],
	["createdList", payload?.createdList],
	["creator", payload?.creator],
	["creator.playlist", payload?.creator?.playlist],
	["creator.playlists", payload?.creator?.playlists],
	["playlist", payload?.playlist],
	["playlists", payload?.playlists]
];
var extractPlaylists = (payload) => {
	debugLog$1("payload top-level keys", Object.keys(payload || {}));
	debugLog$1("payload.data keys", payload?.data && typeof payload.data === "object" ? Object.keys(payload.data) : []);
	const matchedEntry = getNamedCandidateEntries(payload).find(([, candidate]) => Array.isArray(candidate));
	if (matchedEntry) {
		const [candidatePath, playlists] = matchedEntry;
		debugLog$1("matched playlist candidate", {
			candidatePath,
			length: playlists.length
		});
		return playlists;
	}
	debugLog$1("playlist candidates summary", getNamedCandidateEntries(payload).map(([candidatePath, candidate]) => ({
		candidatePath,
		type: Array.isArray(candidate) ? "array" : typeof candidate,
		keys: candidate && typeof candidate === "object" && !Array.isArray(candidate) ? Object.keys(candidate) : void 0
	})));
	throw new Error("用户歌单响应中未找到歌单列表字段");
};
var getErrorMessage = (payload) => {
	return [
		payload.message,
		payload.msg,
		payload.errmsg,
		payload.error
	].find((candidate) => typeof candidate === "string" && candidate.trim() !== "") || "获取用户歌单失败";
};
var getUserPlaylists = async (params) => {
	const { uin, offset = 0, limit = 30, cookie } = params;
	const url = "https://c6.y.qq.com/rsc/fcgi-bin/fcg_get_profile_homepage.fcg";
	const pageOffset = offset % limit;
	try {
		debugLog$1("request meta", {
			url,
			uin,
			offset,
			limit,
			pageOffset,
			hasCookie: Boolean(cookie),
			cookieLength: cookie?.length || 0
		});
		const payload = (await request({
			url,
			method: "GET",
			isUUrl: "u",
			cookie,
			options: {
				params: {
					_: Date.now(),
					cv: 4747474,
					ct: 24,
					format: "json",
					inCharset: "utf-8",
					outCharset: "utf-8",
					notice: 0,
					platform: "yqq.json",
					needNewCode: 0,
					uin: Number.parseInt(uin, 10),
					g_tk_new_20200303: 0,
					g_tk: 0,
					cid: 205360838,
					userid: Number.parseInt(uin, 10),
					reqfrom: 1,
					reqtype: 0,
					hostUin: 0,
					loginUin: Number.parseInt(uin, 10)
				},
				headers: { Referer: `https://y.qq.com/portal/profile.html?uin=${uin}` }
			}
		})).data;
		debugLog$1("upstream payload summary", {
			topLevelKeys: payload && typeof payload === "object" ? Object.keys(payload) : null,
			code: payload?.code,
			hasData: Boolean(payload?.data),
			dataKeys: payload?.data && typeof payload.data === "object" ? Object.keys(payload.data) : []
		});
		if (!payload || typeof payload !== "object") {
			debugLog$1("invalid payload received", payload);
			return errorResponse("用户歌单响应格式无效", 502);
		}
		if (typeof payload.code === "number" && payload.code !== 0) {
			debugLog$1("upstream business error payload", payload);
			return errorResponse(getErrorMessage(payload), 502);
		}
		const upstreamPlaylists = extractPlaylists(payload);
		const playlists = pageOffset > 0 ? upstreamPlaylists.slice(pageOffset, pageOffset + limit) : upstreamPlaylists;
		debugLog$1("final response contract", {
			wrapper: "customResponse",
			expectedBodyShape: { response: {
				code: 0,
				data: { playlists: "..." }
			} },
			upstreamLength: upstreamPlaylists.length,
			normalizedLength: playlists.length
		});
		return customResponse({ response: {
			code: 0,
			data: { playlists }
		} }, 200);
	} catch (error) {
		console.error("获取用户歌单失败:", error);
		return errorResponse(error, 502);
	}
};
//#endregion
//#region src/services/apis/user/getUserAvatar.ts
var getUserAvatar = async (params) => {
	const { k, uin, size = 140 } = params;
	if (k) return {
		avatarUrl: `https://thirdqq.qlogo.cn/g?b=sdk&k=${k}&s=${size}`,
		message: "头像 URL 获取成功"
	};
	if (uin) return {
		avatarUrl: `https://q.qlogo.cn/headimg_dl?dst_uin=${uin}&spec=${size}`,
		message: "头像 URL 获取成功"
	};
	throw new Error("缺少 k 或 uin 参数");
};
//#endregion
//#region src/services/apis/user/getUserLikedSongs.ts
var DEBUG_ENABLED = process.env.DEBUG === "true";
var debugLog = (message, payload) => {
	if (DEBUG_ENABLED) console.log(`[getUserLikedSongs] ${message}`, payload ?? "");
};
var getUserLikedSongs = async (params) => {
	const { uin, offset = 0, limit = 30 } = params;
	const page = Math.floor(offset / limit) + 1;
	const url = "https://c6.y.qq.com/rsc/fcgi-bin/fcg_get_profile_homepage.fcg";
	try {
		const userInfo = require_cookieResolver.getUserInfo();
		debugLog("request meta", {
			url,
			uin,
			offset,
			limit,
			page,
			hasGlobalCookie: Boolean(userInfo.cookie),
			cookieLength: userInfo.cookie?.length || 0
		});
		const payload = (await request({
			url,
			method: "GET",
			isUUrl: "u",
			options: {
				params: {
					_: Date.now(),
					cv: 4747474,
					ct: 24,
					format: "json",
					inCharset: "utf-8",
					outCharset: "utf-8",
					notice: 0,
					platform: "yqq.json",
					needNewCode: 0,
					uin: Number.parseInt(uin, 10),
					g_tk_new_20200303: 0,
					g_tk: 0,
					cid: 205360838,
					userid: Number.parseInt(uin, 10),
					reqfrom: 1,
					reqtype: 0,
					hostUin: 0,
					loginUin: Number.parseInt(uin, 10)
				},
				headers: {
					Referer: `https://y.qq.com/portal/profile.html?uin=${uin}`,
					Cookie: userInfo.cookie || ""
				}
			}
		})).data;
		debugLog("upstream payload summary", {
			topLevelKeys: payload && typeof payload === "object" ? Object.keys(payload) : null,
			code: payload?.code,
			hasData: Boolean(payload?.data),
			dataKeys: payload?.data && typeof payload.data === "object" ? Object.keys(payload.data) : []
		});
		if (!payload || typeof payload !== "object") {
			debugLog("invalid payload received", payload);
			return errorResponse("用户喜欢的歌曲响应格式无效", 502);
		}
		if (typeof payload.code === "number" && payload.code !== 0) {
			debugLog("upstream business error payload", payload);
			return errorResponse(payload.msg || payload.message || "获取用户喜欢的歌曲失败", 502);
		}
		const mymusic = payload?.data?.mymusic;
		let likedSongsInfo = null;
		if (Array.isArray(mymusic)) likedSongsInfo = mymusic.find((item) => {
			return item?.title && (item.title.includes("喜欢") || item.type === 1);
		});
		debugLog("liked songs info", likedSongsInfo);
		if (!likedSongsInfo) {
			debugLog("no liked songs info found in mymusic");
			return customResponse({ response: {
				code: 0,
				data: {
					songs: [],
					total: 0,
					hasMore: false
				}
			} }, 200);
		}
		return customResponse({ response: {
			code: 0,
			data: {
				songs: [likedSongsInfo],
				total: likedSongsInfo.num0 || 0,
				hasMore: false,
				info: {
					title: likedSongsInfo.title,
					songCount: likedSongsInfo.num0,
					albumCount: likedSongsInfo.num1,
					playlistCount: likedSongsInfo.num2,
					id: likedSongsInfo.id
				}
			}
		} }, 200);
	} catch (error) {
		console.error("获取用户喜欢的歌曲失败:", error);
		return errorResponse(error, 502);
	}
};
//#endregion
//#region src/services/apis/user/getUserDetail.ts
var getUserDetail = async ({ uin, cookie }) => handleApi(request({
	url: "/rsc/fcgi-bin/fcg_get_profile_homepage.fcg",
	method: "GET",
	cookie,
	options: {
		params: {
			_: Date.now(),
			cv: 4747474,
			ct: 24,
			format: "json",
			inCharset: "utf-8",
			outCharset: "utf-8",
			notice: 0,
			platform: "yqq.json",
			needNewCode: 0,
			uin: Number.parseInt(uin, 10),
			g_tk_new_20200303: 0,
			g_tk: 0,
			cid: 205360838,
			userid: Number.parseInt(uin, 10),
			reqfrom: 1
		},
		headers: { Referer: `https://y.qq.com/portal/profile.html?uin=${uin}` }
	}
}));
//#endregion
//#region src/services/apis/user/getUserCollections.ts
var getUserCollection = async ({ uin, type, page = 1, limit = 20, cookie }) => {
	return handleApi(request({
		url: "/fav/fcgi-bin/fcg_get_profile_order_asset.fcg",
		method: "GET",
		cookie,
		options: {
			params: {
				ct: 20,
				cid: 205360956,
				userid: uin,
				reqtype: type === "songlist" ? 3 : 2,
				sin: (page - 1) * limit,
				ein: type === "songlist" ? page * limit : page * limit - 1,
				format: "json",
				g_tk: 5381
			},
			headers: { Referer: `https://y.qq.com/portal/profile.html?uin=${uin}` }
		}
	}));
};
var getUserCollectedSongLists = (params) => getUserCollection({
	...params,
	type: "songlist"
});
var getUserCollectedAlbums = (params) => getUserCollection({
	...params,
	type: "album"
});
//#endregion
//#region src/services/apis/user/getUserSocial.ts
var getUserFollowSingers = async ({ uin, page = 1, limit = 20, cookie }) => handleApi(request({
	url: "/rsc/fcgi-bin/fcg_order_singer_getlist.fcg",
	method: "GET",
	cookie,
	options: { params: {
		utf8: 1,
		page,
		perpage: limit,
		uin,
		g_tk: 5381,
		format: "json"
	} }
}));
var getUserFollowOrFans = async ({ uin, page = 1, limit = 20, cookie, isListen }) => handleApi(request({
	url: "/rsc/fcgi-bin/friend_follow_or_listen_list.fcg",
	method: "GET",
	cookie,
	options: { params: {
		utf8: 1,
		start: (page - 1) * limit,
		num: limit,
		uin,
		format: "json",
		g_tk: 5381,
		...isListen ? { is_listen: 1 } : {}
	} }
}));
var getUserFollowUsers = (params) => getUserFollowOrFans(params);
var getUserFans = (params) => getUserFollowOrFans({
	...params,
	isListen: true
});
//#endregion
exports.UCommon = UCommon_default;
exports.UCommon_default = UCommon_default;
exports.checkQQLoginQr = checkQQLoginQr;
exports.checkQQLoginQr$1 = checkQQLoginQr;
exports.colors = colors;
exports.customResponse = customResponse;
exports.downloadQQMusic = downloadQQMusic_default;
exports.downloadQQMusic_default = downloadQQMusic_default;
exports.errorResponse = errorResponse;
exports.getAlbumInfo = getAlbumInfo_default;
exports.getAlbumInfo_default = getAlbumInfo_default;
exports.getAlbumSongs = getAlbumSongs_default;
exports.getAlbumSongs_default = getAlbumSongs_default;
exports.getComments = getComments_default;
exports.getComments_default = getComments_default;
exports.getCommonParams = getCommonParams;
exports.getDigitalAlbumLists = getDigitalAlbumLists_default;
exports.getDigitalAlbumLists_default = getDigitalAlbumLists_default;
exports.getHotKey = getHotKey_default;
exports.getHotKey_default = getHotKey_default;
exports.getLyric = getLyric_default;
exports.getLyric_default = getLyric_default;
exports.getMusicPlay = getMusicPlay_default;
exports.getMusicPlay_default = getMusicPlay_default;
exports.getMvByTag = getMvByTag_default;
exports.getMvByTag_default = getMvByTag_default;
exports.getMvCategory = getMvCategory_default;
exports.getMvCategory_default = getMvCategory_default;
exports.getQQLoginQr = getQQLoginQr;
exports.getQQLoginQr$1 = getQQLoginQr;
exports.getRadioLists = getRadioLists_default;
exports.getRadioLists_default = getRadioLists_default;
exports.getRecommendBanner = getRecommendBanner_default;
exports.getRecommendBanner_default = getRecommendBanner_default;
exports.getRelatedMv = getRelatedMv_default;
exports.getRelatedMv_default = getRelatedMv_default;
exports.getRelatedPlaylists = getRelatedPlaylists_default;
exports.getRelatedPlaylists_default = getRelatedPlaylists_default;
exports.getSearchByKey = getSearchByKey_default;
exports.getSearchByKey_default = getSearchByKey_default;
exports.getSimilarSinger = getSimilarSinger_default;
exports.getSimilarSinger_default = getSimilarSinger_default;
exports.getSingerCategory = getSingerCategory_default;
exports.getSingerCategory_default = getSingerCategory_default;
exports.getSingerDesc = getSingerDesc_default;
exports.getSingerDesc_default = getSingerDesc_default;
exports.getSingerMv = getSingerMv_default;
exports.getSingerMv_default = getSingerMv_default;
exports.getSingerStarNum = getSingerStarNum_default;
exports.getSingerStarNum_default = getSingerStarNum_default;
exports.getSmartbox = getSmartbox_default;
exports.getSmartbox_default = getSmartbox_default;
exports.getTopLists = getTopLists_default;
exports.getTopLists_default = getTopLists_default;
exports.getUserAvatar = getUserAvatar;
exports.getUserAvatar$1 = getUserAvatar;
exports.getUserCollectedAlbums = getUserCollectedAlbums;
exports.getUserCollectedAlbums$1 = getUserCollectedAlbums;
exports.getUserCollectedSongLists = getUserCollectedSongLists;
exports.getUserCollectedSongLists$1 = getUserCollectedSongLists;
exports.getUserDetail = getUserDetail;
exports.getUserDetail$1 = getUserDetail;
exports.getUserFans = getUserFans;
exports.getUserFans$1 = getUserFans;
exports.getUserFollowSingers = getUserFollowSingers;
exports.getUserFollowSingers$1 = getUserFollowSingers;
exports.getUserFollowUsers = getUserFollowUsers;
exports.getUserFollowUsers$1 = getUserFollowUsers;
exports.getUserLikedSongs = getUserLikedSongs;
exports.getUserLikedSongs$1 = getUserLikedSongs;
exports.getUserPlaylists = getUserPlaylists;
exports.getUserPlaylists$1 = getUserPlaylists;
exports.handleApi = handleApi;
exports.logControllerFailure = logControllerFailure;
exports.logControllerStart = logControllerStart;
exports.logControllerSuccess = logControllerSuccess;
exports.songListCategories = songListCategories_default;
exports.songListCategories_default = songListCategories_default;
exports.songListDetail = songListDetail_default;
exports.songListDetail_default = songListDetail_default;
exports.songLists = songLists_default;
exports.songLists_default = songLists_default;
exports.startTimer = startTimer;
exports.u_common_default = u_common_default;
