// @sansenjian/qq-music-api/dist/server-CdzMPnXx.cjs
const require_cookieResolver = require("./cookieResolver-In4BnxUe.cjs");
const require_services = require("./services.cjs");
const require_package = require("./package-8fvg49I3.cjs");
let node_fs = require("node:fs");
node_fs = require_cookieResolver.__toESM(node_fs, 1);
let node_path = require("node:path");
node_path = require_cookieResolver.__toESM(node_path, 1);
let node_url = require("node:url");
let koa = require("koa");
koa = require_cookieResolver.__toESM(koa, 1);
let fs = require("fs");
fs = require_cookieResolver.__toESM(fs, 1);
let node_fs_promises = require("node:fs/promises");
node_fs_promises = require_cookieResolver.__toESM(node_fs_promises, 1);
let _koa_router = require("@koa/router");
let node_util = require("node:util");
//#region src/middlewares/body-parser.ts
var METHODS = new Set([
	"POST",
	"PUT",
	"PATCH"
]);
var JSON_BODY_LIMIT = 1024 * 1024;
var FORM_BODY_LIMIT = 56 * 1024;
var getBodyLimit = (contentType) => contentType.includes("application/x-www-form-urlencoded") ? FORM_BODY_LIMIT : JSON_BODY_LIMIT;
var parseUrlEncoded = (raw) => {
	const body = {};
	new URLSearchParams(raw).forEach((value, key) => {
		const current = body[key];
		if (current === void 0) body[key] = value;
		else if (Array.isArray(current)) current.push(value);
		else body[key] = [current, value];
	});
	return body;
};
var bodyParser = () => async (ctx, next) => {
	if (!METHODS.has(ctx.method)) return await next();
	const contentType = ctx.get("Content-Type") || "";
	const bodyLimit = getBodyLimit(contentType);
	const buffers = [];
	let bodySize = 0;
	for await (const chunk of ctx.req) {
		const buffer = typeof chunk === "string" ? Buffer.from(chunk) : chunk;
		bodySize += buffer.byteLength;
		if (bodySize > bodyLimit) ctx.throw(413, "Request body too large");
		buffers.push(buffer);
	}
	if (buffers.length === 0) return await next();
	const raw = Buffer.concat(buffers).toString("utf-8");
	try {
		if (contentType.includes("application/json")) ctx.request.body = JSON.parse(raw);
		else if (contentType.includes("application/x-www-form-urlencoded")) ctx.request.body = parseUrlEncoded(raw);
		else ctx.request.body = raw;
	} catch {
		ctx.throw(400, "Invalid JSON in request body");
	}
	await next();
};
//#endregion
//#region src/middlewares/static-serve.ts
var MIME_TYPES = {
	".html": "text/html; charset=utf-8",
	".css": "text/css; charset=utf-8",
	".js": "application/javascript; charset=utf-8",
	".json": "application/json; charset=utf-8",
	".png": "image/png",
	".jpg": "image/jpeg",
	".jpeg": "image/jpeg",
	".svg": "image/svg+xml",
	".ico": "image/x-icon",
	".woff": "font/woff",
	".woff2": "font/woff2"
};
var isInsideRoot = (root, filePath) => {
	const relativePath = node_path.default.relative(root, filePath);
	return relativePath === "" || !relativePath.startsWith("..") && !node_path.default.isAbsolute(relativePath);
};
var staticServe = (root) => async (ctx, next) => {
	if (ctx.method !== "GET" && ctx.method !== "HEAD") return await next();
	const rootPath = node_path.default.resolve(root);
	const safePath = node_path.default.normalize(ctx.path.slice(1)).replace(/^(\.\.[/\\])+/, "");
	let filePath = node_path.default.resolve(rootPath, safePath);
	if (!isInsideRoot(rootPath, filePath) || !node_fs.default.existsSync(filePath)) return await next();
	let stat = node_fs.default.statSync(filePath);
	if (stat.isDirectory()) {
		filePath = node_path.default.join(filePath, "index.html");
		if (!isInsideRoot(rootPath, filePath) || !node_fs.default.existsSync(filePath)) return await next();
		stat = node_fs.default.statSync(filePath);
	}
	if (!stat.isFile()) return await next();
	ctx.type = MIME_TYPES[node_path.default.extname(filePath).toLowerCase()] || "application/octet-stream";
	ctx.length = stat.size;
	ctx.body = node_fs.default.createReadStream(filePath);
};
//#endregion
//#region src/middlewares/koa-cors.ts
function resolveOrigin(ctx, options, credentialsEnabled) {
	const requestOrigin = ctx.get("Origin");
	if (typeof options.origin === "function") {
		const computedOrigin = options.origin(ctx);
		if (credentialsEnabled && computedOrigin === "null") return "";
		return computedOrigin;
	}
	if (Array.isArray(options.origin)) return requestOrigin && options.origin.includes(requestOrigin) ? requestOrigin : "";
	if (typeof options.origin === "string") {
		if (credentialsEnabled && options.origin === "null") return "";
		return options.origin;
	}
	return "*";
}
function crossOrigin(options = {}) {
	options = Object.assign({}, { allowMethods: [
		"GET",
		"PUT",
		"POST",
		"PATCH",
		"DELETE",
		"HEAD",
		"OPTIONS"
	] }, options);
	return async function cors(ctx, next) {
		ctx.vary("Origin");
		const origin = resolveOrigin(ctx, options, options.credentials === true);
		if (!origin) return await next();
		ctx.set("Access-Control-Allow-Origin", origin);
		if (ctx.method === "OPTIONS") {
			if (!ctx.get("Access-Control-Request-Method")) return await next();
			if (options.maxAge) ctx.set("Access-Control-Max-Age", String(options.maxAge));
			if (options.credentials === true && origin !== "*") ctx.set("Access-Control-Allow-Credentials", "true");
			if (options.allowMethods) ctx.set("Access-Control-Allow-Methods", options.allowMethods.join(","));
			if (options.allowHeaders) ctx.set("Access-Control-Allow-Headers", options.allowHeaders.join(","));
			else ctx.set("Access-Control-Allow-Headers", ctx.get("Access-Control-Request-Headers"));
			ctx.status = 204;
		} else {
			if (options.credentials === true && origin !== "*") ctx.set("Access-Control-Allow-Credentials", "true");
			if (options.exposeHeaders) ctx.set("Access-Control-Expose-Headers", options.exposeHeaders.join(","));
			try {
				await next();
			} catch (err) {
				throw err;
			}
		}
	};
}
//#endregion
//#region src/middlewares/security-headers.ts
var securityHeaders = () => async (ctx, next) => {
	ctx.set("X-Content-Type-Options", "nosniff");
	ctx.set("X-Frame-Options", "DENY");
	ctx.set("Referrer-Policy", "no-referrer");
	await next();
};
//#endregion
//#region src/routes/api-metadata.ts
var apiMetadata = [
	{
		name: "getCookie",
		category: "user",
		method: "GET",
		path: "/user/getCookie"
	},
	{
		name: "setCookie",
		category: "user",
		method: "GET",
		path: "/user/setCookie",
		queryParams: [{
			name: "cookie",
			required: true
		}]
	},
	{
		name: "getUserPlaylists",
		category: "user",
		method: "GET",
		path: "/user/getUserPlaylists",
		queryParams: [
			{
				name: "uin",
				required: true
			},
			{ name: "offset" },
			{ name: "limit" },
			{ name: "cookie" }
		],
		cookieRequired: true
	},
	{
		name: "getUserDetail",
		category: "user",
		method: "GET",
		path: "/user/getUserDetail",
		queryParams: [{
			name: "uin",
			required: true
		}, { name: "cookie" }],
		cookieRequired: true
	},
	{
		name: "getUserCollectedSongLists",
		category: "user",
		method: "GET",
		path: "/user/getUserCollectedSongLists",
		queryParams: [
			{
				name: "uin",
				required: true
			},
			{ name: "page" },
			{ name: "limit" },
			{ name: "cookie" }
		],
		cookieRequired: true
	},
	{
		name: "getUserCollectedAlbums",
		category: "user",
		method: "GET",
		path: "/user/getUserCollectedAlbums",
		queryParams: [
			{
				name: "uin",
				required: true
			},
			{ name: "page" },
			{ name: "limit" },
			{ name: "cookie" }
		],
		cookieRequired: true
	},
	{
		name: "getUserFollowSingers",
		category: "user",
		method: "GET",
		path: "/user/getUserFollowSingers",
		queryParams: [
			{
				name: "uin",
				required: true
			},
			{ name: "page" },
			{ name: "limit" },
			{ name: "cookie" }
		],
		cookieRequired: true
	},
	{
		name: "getUserFollowUsers",
		category: "user",
		method: "GET",
		path: "/user/getUserFollowUsers",
		queryParams: [
			{
				name: "uin",
				required: true
			},
			{ name: "page" },
			{ name: "limit" },
			{ name: "cookie" }
		],
		cookieRequired: true
	},
	{
		name: "getUserFans",
		category: "user",
		method: "GET",
		path: "/user/getUserFans",
		queryParams: [
			{
				name: "uin",
				required: true
			},
			{ name: "page" },
			{ name: "limit" },
			{ name: "cookie" }
		],
		cookieRequired: true
	},
	{
		name: "getUserAvatar",
		category: "user",
		method: "GET",
		path: "/user/getUserAvatar",
		queryParams: [
			{ name: "uin" },
			{ name: "k" },
			{ name: "size" }
		]
	},
	{
		name: "getUserLikedSongs",
		category: "user",
		method: "GET",
		path: "/user/getUserLikedSongs",
		queryParams: [
			{
				name: "uin",
				required: true
			},
			{ name: "offset" },
			{ name: "limit" },
			{ name: "cookie" }
		],
		cookieRequired: true
	},
	{
		name: "getDownloadQQMusic",
		category: "download",
		method: "GET",
		path: "/downloadQQMusic"
	},
	{
		name: "getHotKey",
		category: "search",
		method: "GET",
		path: "/getHotkey"
	},
	{
		name: "getSearchByKey",
		category: "search",
		method: "GET",
		path: "/getSearchByKey",
		aliases: ["/getSearchByKey/:key"],
		queryParams: [
			{
				name: "key",
				required: true
			},
			{ name: "limit" },
			{ name: "page" },
			{ name: "catZhida" },
			{ name: "remoteplace" }
		]
	},
	{
		name: "getSmartbox",
		category: "search",
		method: "GET",
		path: "/getSmartbox",
		aliases: ["/getSmartbox/:key"],
		queryParams: [{
			name: "key",
			required: true
		}]
	},
	{
		name: "getSongListCategories",
		category: "playlist",
		method: "GET",
		path: "/getSongListCategories"
	},
	{
		name: "getSongLists",
		category: "playlist",
		method: "GET",
		path: "/getSongLists",
		aliases: ["/getSongLists/:page/:limit/:categoryId/:sortId"],
		queryParams: [
			{ name: "limit" },
			{ name: "page" },
			{ name: "sortId" },
			{ name: "categoryId" }
		]
	},
	{
		name: "batchGetSongLists",
		category: "playlist",
		method: "POST",
		path: "/batchGetSongLists",
		bodyExample: { ids: ["123"] }
	},
	{
		name: "getSongInfo",
		category: "music",
		method: "GET",
		path: "/getSongInfo",
		aliases: ["/getSongInfo/:songmid"],
		queryParams: [{
			name: "songmid",
			required: true
		}, { name: "songid" }]
	},
	{
		name: "batchGetSongInfo",
		category: "music",
		method: "POST",
		path: "/batchGetSongInfo",
		bodyExample: { songs: [["songmid", "songid"]] }
	},
	{
		name: "getSongListDetail",
		category: "playlist",
		method: "GET",
		path: "/getSongListDetail",
		aliases: ["/getSongListDetail/:disstid"],
		queryParams: [{
			name: "disstid",
			required: true
		}]
	},
	{
		name: "getAlbumSongs",
		category: "album",
		method: "GET",
		path: "/getAlbumSongs",
		aliases: ["/getAlbumSongs/:albummid"],
		queryParams: [
			{
				name: "albummid",
				required: true
			},
			{ name: "albumid" },
			{ name: "begin" },
			{ name: "limit" },
			{ name: "order" }
		]
	},
	{
		name: "getNewDisks",
		category: "album",
		method: "GET",
		path: "/getNewDisks",
		queryParams: [{ name: "page" }, { name: "limit" }]
	},
	{
		name: "getMvByTag",
		category: "mv",
		method: "GET",
		path: "/getMvByTag"
	},
	{
		name: "getMv",
		category: "mv",
		method: "GET",
		path: "/getMv",
		queryParams: [
			{ name: "area_id" },
			{ name: "version_id" },
			{ name: "limit" },
			{ name: "page" }
		]
	},
	{
		name: "getMvCategory",
		category: "mv",
		method: "GET",
		path: "/getMvCategory"
	},
	{
		name: "getMvPlay",
		category: "mv",
		method: "GET",
		path: "/getMvPlay",
		queryParams: [{
			name: "vid",
			required: true
		}]
	},
	{
		name: "getSingerList",
		category: "singer",
		method: "GET",
		path: "/getSingerList",
		queryParams: [
			{ name: "area" },
			{ name: "sex" },
			{ name: "genre" },
			{ name: "index" },
			{ name: "page" }
		]
	},
	{
		name: "getSingerCategory",
		category: "singer",
		method: "GET",
		path: "/getSingerCategory"
	},
	{
		name: "getSimilarSinger",
		category: "singer",
		method: "GET",
		path: "/getSimilarSinger",
		queryParams: [{
			name: "singermid",
			required: true
		}]
	},
	{
		name: "getSingerAlbum",
		category: "singer",
		method: "GET",
		path: "/getSingerAlbum",
		queryParams: [
			{
				name: "singermid",
				required: true
			},
			{ name: "limit" },
			{ name: "page" }
		]
	},
	{
		name: "getSingerHotsong",
		category: "singer",
		method: "GET",
		path: "/getSingerHotsong",
		queryParams: [
			{
				name: "singermid",
				required: true
			},
			{ name: "limit" },
			{ name: "page" }
		]
	},
	{
		name: "getSingerMv",
		category: "singer",
		method: "GET",
		path: "/getSingerMv",
		queryParams: [
			{
				name: "singermid",
				required: true
			},
			{ name: "order" },
			{ name: "num" }
		]
	},
	{
		name: "getSingerDesc",
		category: "singer",
		method: "GET",
		path: "/getSingerDesc",
		queryParams: [{
			name: "singermid",
			required: true
		}]
	},
	{
		name: "getSingerStarNum",
		category: "singer",
		method: "GET",
		path: "/getSingerStarNum",
		queryParams: [{
			name: "singermid",
			required: true
		}]
	},
	{
		name: "getRadioLists",
		category: "radio",
		method: "GET",
		path: "/getRadioLists"
	},
	{
		name: "getDigitalAlbumLists",
		category: "album",
		method: "GET",
		path: "/getDigitalAlbumLists"
	},
	{
		name: "getLyric",
		category: "music",
		method: "GET",
		path: "/getLyric",
		aliases: ["/getLyric/:songmid"],
		queryParams: [
			{
				name: "songmid",
				required: true
			},
			{ name: "isFormat" },
			{ name: "cookie" }
		],
		cookieRequired: true
	},
	{
		name: "getMusicPlay",
		category: "music",
		method: "GET",
		path: "/getMusicPlay",
		aliases: ["/getMusicPlay/:songmid"],
		queryParams: [
			{
				name: "songmid",
				required: true
			},
			{ name: "resType" },
			{ name: "mediaId" },
			{ name: "quality" },
			{ name: "cookie" }
		],
		cookieRequired: true
	},
	{
		name: "getAlbumInfo",
		category: "album",
		method: "GET",
		path: "/getAlbumInfo",
		aliases: ["/getAlbumInfo/:albummid"],
		queryParams: [{
			name: "albummid",
			required: true
		}]
	},
	{
		name: "getComments",
		category: "comments",
		method: "GET",
		path: "/getComments",
		queryParams: [
			{
				name: "id",
				required: true
			},
			{ name: "pagesize" },
			{ name: "pagenum" },
			{ name: "cid" },
			{ name: "cmd" },
			{ name: "reqtype" },
			{ name: "biztype" },
			{ name: "rootcommentid" }
		]
	},
	{
		name: "getRecommend",
		category: "recommend",
		method: "GET",
		path: "/getRecommend"
	},
	{
		name: "getRecommendBanner",
		category: "recommend",
		method: "GET",
		path: "/getRecommendBanner"
	},
	{
		name: "getTopLists",
		category: "rank",
		method: "GET",
		path: "/getTopLists"
	},
	{
		name: "getRanks",
		category: "rank",
		method: "GET",
		path: "/getRanks"
	},
	{
		name: "getTicketInfo",
		category: "utility",
		method: "GET",
		path: "/getTicketInfo"
	},
	{
		name: "getImageUrl",
		category: "utility",
		method: "GET",
		path: "/getImageUrl",
		queryParams: [
			{
				name: "id",
				required: true
			},
			{ name: "size" },
			{ name: "maxAge" }
		]
	},
	{
		name: "getQQLoginQr",
		category: "login",
		method: "GET",
		path: "/getQQLoginQr",
		aliases: ["/user/getQQLoginQr"]
	},
	{
		name: "checkQQLoginQr",
		category: "login",
		method: "POST",
		path: "/checkQQLoginQr",
		aliases: ["/user/checkQQLoginQr"]
	},
	{
		name: "getDailyRecommend",
		category: "recommend",
		method: "GET",
		path: "/getDailyRecommend",
		cookieRequired: true
	},
	{
		name: "getPrivateFM",
		category: "recommend",
		method: "GET",
		path: "/getPrivateFM",
		cookieRequired: true
	},
	{
		name: "getNewSongs",
		category: "recommend",
		method: "GET",
		path: "/getNewSongs",
		queryParams: [{ name: "areaId" }, { name: "limit" }]
	},
	{
		name: "getPersonalRecommend",
		category: "recommend",
		method: "GET",
		path: "/getPersonalRecommend",
		queryParams: [{ name: "type" }, { name: "cookie" }],
		cookieRequired: true
	},
	{
		name: "getSimilarSongs",
		category: "recommend",
		method: "GET",
		path: "/getSimilarSongs",
		queryParams: [{
			name: "songmid",
			required: true
		}]
	},
	{
		name: "getRelatedPlaylists",
		category: "recommend",
		method: "GET",
		path: "/getRelatedPlaylists",
		queryParams: [
			{
				name: "songid",
				required: true
			},
			{ name: "sin" },
			{ name: "lastId" },
			{ name: "songType" }
		]
	},
	{
		name: "getRelatedMv",
		category: "recommend",
		method: "GET",
		path: "/getRelatedMv",
		queryParams: [
			{
				name: "songid",
				required: true
			},
			{ name: "songtype" },
			{ name: "lastmvid" },
			{ name: "limit" }
		]
	},
	{
		name: "getPlaylistTags",
		category: "playlist",
		method: "GET",
		path: "/getPlaylistTags"
	},
	{
		name: "getPlaylistsByTag",
		category: "playlist",
		method: "GET",
		path: "/getPlaylistsByTag",
		queryParams: [
			{ name: "tagId" },
			{ name: "page" },
			{ name: "num" }
		]
	},
	{
		name: "getHotComments",
		category: "comments",
		method: "GET",
		path: "/getHotComments",
		queryParams: [
			{
				name: "id",
				required: true
			},
			{ name: "type" },
			{ name: "page" },
			{ name: "pagesize" }
		]
	},
	{
		name: "getSingerListByArea",
		category: "singer",
		method: "GET",
		path: "/getSingerListByArea",
		queryParams: [
			{ name: "area" },
			{ name: "sex" },
			{ name: "genre" },
			{ name: "page" },
			{ name: "pagesize" }
		]
	}
];
apiMetadata.flatMap((item) => [item.path, ...item.aliases || []]);
var API_EXPLORER_INDEX_PATH = "/explorer/index.html";
var API_EXPLORER_METADATA_PATH = "/explorer/metadata";
var METADATA_PATH_PLACEHOLDER = "__API_EXPLORER_METADATA_PATH__";
var renderExplorerIndex = async (publicDir) => {
	const templatePath = node_path.default.join(publicDir, "explorer", "index.html");
	return (await node_fs_promises.default.readFile(templatePath, "utf8")).split(METADATA_PATH_PLACEHOLDER).join(API_EXPLORER_METADATA_PATH);
};
var apiExplorer = (options = {}) => async (ctx, next) => {
	if (ctx.method !== "GET" && ctx.method !== "HEAD") {
		await next();
		return;
	}
	if (ctx.path === "/explorer") {
		ctx.redirect(API_EXPLORER_INDEX_PATH);
		return;
	}
	if (ctx.path === "/explorer/index.html" && options.publicDir) {
		ctx.type = "text/html; charset=utf-8";
		ctx.body = await renderExplorerIndex(options.publicDir);
		return;
	}
	if (ctx.path === "/explorer/metadata") {
		ctx.type = "application/json";
		ctx.body = {
			title: "QQ Music API Explorer",
			description: "Local API request explorer powered by the server metadata.",
			endpoints: apiMetadata
		};
		return;
	}
	await next();
};
//#endregion
//#region src/controllers/util.ts
var INTERNAL_ERROR_MESSAGE = "服务器内部错误";
var UPSTREAM_ERROR_MESSAGE = "上游服务异常";
var setJsonResponse = (ctx, status, body) => {
	ctx.status = status;
	ctx.type = "application/json";
	ctx.body = body;
};
var isMissingRequiredValue = (value) => {
	if (value === void 0 || value === null) return true;
	if (typeof value === "string") return value.trim() === "";
	return false;
};
function createController(apiFunction, options) {
	return async (ctx, _next) => {
		const controllerName = options?.name || apiFunction.name || "anonymous";
		const duration = require_services.startTimer();
		require_services.logControllerStart(controllerName, ctx);
		try {
			const params = {
				...ctx.query,
				...ctx.params
			};
			if (options?.validator) {
				const validation = options.validator(params);
				if (!validation.valid) {
					setJsonResponse(ctx, 400, { response: validation.error || options.errorMessage || "Invalid parameters" });
					return;
				}
			}
			const { status, body } = await apiFunction(options?.buildApiOptions ? options.buildApiOptions(ctx, params) : {
				method: "get",
				params,
				option: {}
			});
			setJsonResponse(ctx, status, body);
			require_services.logControllerSuccess(controllerName, ctx, duration());
		} catch (error) {
			require_services.logControllerFailure(controllerName, error, duration());
			if (options?.onError) options.onError(ctx, error);
			else setJsonResponse(ctx, 500, { error: INTERNAL_ERROR_MESSAGE });
		}
	};
}
function validateRequired(fields) {
	return (params) => {
		const missingFields = fields.filter((field) => isMissingRequiredValue(params[field]));
		if (missingFields.length > 0) return {
			valid: false,
			error: `缺少必需参数：${missingFields.join(", ")}`
		};
		return { valid: true };
	};
}
function setApiResponse(ctx, apiResponse) {
	setJsonResponse(ctx, apiResponse.status || 500, apiResponse.body);
}
function withErrorHandler(handler, name = handler.name || "anonymous") {
	return async (ctx, next) => {
		const duration = require_services.startTimer();
		require_services.logControllerStart(name, ctx);
		try {
			await handler(ctx);
			require_services.logControllerSuccess(name, ctx, duration());
			await next();
		} catch (error) {
			require_services.logControllerFailure(name, error, duration());
			setJsonResponse(ctx, 502, { error: UPSTREAM_ERROR_MESSAGE });
		}
	};
}
//#endregion
//#region src/controllers/getDownloadQQMusic.ts
var getDownloadQQMusic_default = createController(require_services.downloadQQMusic_default);
//#endregion
//#region src/controllers/getHotkey.ts
var getHotkey_default = createController(require_services.getHotKey_default);
//#endregion
//#region src/controllers/getSearchByKey.ts
var getSearchByKey_default = withErrorHandler(async (ctx) => {
	const w = ctx.query.key || ctx.params.key;
	const { limit: n, page: p, catZhida, remoteplace = "song" } = ctx.query;
	if (!w) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "search key is null" }
		});
		return;
	}
	setApiResponse(ctx, await require_services.getSearchByKey_default({
		method: "get",
		params: {
			w,
			n: +n || 10,
			p: +p || 1,
			catZhida: +catZhida || 1,
			remoteplace: `txt.yqq.${remoteplace}`
		},
		option: {}
	}));
}, "getSearchByKey");
//#endregion
//#region src/controllers/getSmartbox.ts
var getSmartbox_default = withErrorHandler(async (ctx) => {
	const { key } = ctx.query;
	if (!key) {
		setApiResponse(ctx, {
			status: 200,
			body: { response: null }
		});
		return;
	}
	setApiResponse(ctx, await require_services.getSmartbox_default({
		method: "get",
		params: { key },
		option: {}
	}));
});
//#endregion
//#region src/controllers/getSongListCategories.ts
/**
* @description: 歌单
* 1 歌单类型
* 2 所有歌单
* 3 分类歌单
* 4 歌单详情
*
*/
var getSongListCategoriesController = withErrorHandler(async (ctx) => {
	setApiResponse(ctx, await require_services.songListCategories_default({
		method: "get",
		params: {},
		option: {}
	}));
});
//#endregion
//#region src/controllers/getSongLists.ts
var getSongLists_default = withErrorHandler(async (ctx) => {
	const { limit = 20, page = 0, sortId = 5, categoryId = 1e7 } = ctx.query;
	setApiResponse(ctx, await require_services.songLists_default({
		method: "get",
		params: {
			categoryId,
			sortId,
			sin: +page * +limit,
			ein: +limit * (+page + 1) - 1
		},
		option: {}
	}));
});
//#endregion
//#region src/controllers/batchGetSongLists.ts
var isSongListResponse = (value) => Boolean(value && typeof value === "object");
var batchGetSongListsController = withErrorHandler(async (ctx) => {
	const { limit: ein = 19, page: sin = 0, sortId = 5, categoryIds = [1e7] } = ctx.request.body || {};
	const params = {
		sortId,
		sin,
		ein
	};
	const props = {
		method: "get",
		option: {},
		params
	};
	setApiResponse(ctx, require_services.customResponse({
		status: 200,
		data: await Promise.all(categoryIds.map(async (categoryId) => {
			const response = (await require_services.songLists_default({
				...props,
				params: {
					...params,
					categoryId
				}
			})).body.response;
			if (isSongListResponse(response) && Number(response.code) === 0) return response.data;
			else return response;
		}))
	}, 200));
});
//#endregion
//#region src/controllers/getSongInfo.ts
var getSongInfoController = withErrorHandler(async (ctx) => {
	setApiResponse(ctx, require_services.customResponse({ response: (await require_services.UCommon_default({
		method: "get",
		params: {
			format: "json",
			inCharset: "utf8",
			outCharset: "utf-8",
			notice: 0,
			platform: "yqq.json",
			needNewCode: 0,
			data: {
				comm: {
					ct: 24,
					cv: 0
				},
				songinfo: {
					method: "get_song_detail_yqq",
					param: {
						song_type: 0,
						song_mid: ctx.query.songmid,
						song_id: ctx.query.songid || ""
					},
					module: "music.pf_song_detail_svr"
				}
			}
		},
		option: {}
	})).data }, 200));
});
//#endregion
//#region src/controllers/batchGetSongInfo.ts
var batchGetSongInfoController = withErrorHandler(async (ctx) => {
	const { songs } = ctx.request.body || {};
	const params = {
		format: "json",
		inCharset: "utf8",
		outCharset: "utf-8",
		notice: 0,
		platform: "yqq.json",
		needNewCode: 0
	};
	const props = {
		method: "get",
		option: {},
		params
	};
	setApiResponse(ctx, require_services.customResponse({ response: {
		code: 0,
		data: await Promise.all((songs || []).map(async (song) => {
			const [song_mid, song_id = ""] = song;
			return (await require_services.UCommon_default({
				...props,
				params: {
					...params,
					data: {
						comm: {
							ct: 24,
							cv: 0
						},
						songinfo: {
							method: "get_song_detail_yqq",
							param: {
								song_type: 0,
								song_mid,
								song_id
							},
							module: "music.pf_song_detail_svr"
						}
					}
				}
			})).data;
		}))
	} }, 200));
});
//#endregion
//#region src/controllers/getSongListDetail.ts
var getSongListDetailController = withErrorHandler(async (ctx) => {
	const disstid = ctx.query.disstid || ctx.params.disstid;
	if (!disstid) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "no disstid" }
		});
		return;
	}
	setApiResponse(ctx, await require_services.songListDetail_default({
		method: "get",
		params: { disstid },
		option: {}
	}));
});
//#endregion
//#region src/controllers/getAlbumSongs.ts
var getAlbumSongs_default = withErrorHandler(async (ctx) => {
	const albummid = ctx.query.albummid || ctx.params.albummid;
	if (!albummid) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "no albummid" }
		});
		return;
	}
	setApiResponse(ctx, await require_services.getAlbumSongs_default({
		method: "post",
		params: {
			albummid,
			albumid: ctx.query.albumid,
			begin: ctx.query.begin,
			limit: ctx.query.limit,
			order: ctx.query.order
		},
		option: {}
	}));
}, "getAlbumSongs");
//#endregion
//#region src/controllers/getNewDisks.ts
var getNewDisks_default = withErrorHandler(async (ctx) => {
	const page = +ctx.query.page || 1;
	const num = +ctx.query.limit || 20;
	const start = (page - 1) * num;
	const data = {
		new_album: {
			module: "newalbum.NewAlbumServer",
			method: "get_new_album_info",
			param: {
				area: 1,
				start,
				num
			}
		},
		comm: {
			ct: 24,
			cv: 0
		}
	};
	if (!start) data.new_album_tag = {
		module: "newalbum.NewAlbumServer",
		method: "get_new_album_area",
		param: {}
	};
	setApiResponse(ctx, {
		status: 200,
		body: { response: (await require_services.UCommon_default({
			method: "get",
			params: {
				format: "json",
				data: JSON.stringify(data)
			},
			option: {}
		})).data }
	});
});
//#endregion
//#region src/controllers/getMvByTag.ts
var getMvByTag_default = createController(require_services.getMvByTag_default);
//#endregion
//#region src/controllers/getMv.ts
var getMvController = withErrorHandler(async (ctx) => {
	const { area_id = 15, version_id = 7, limit = 20, page = 0 } = ctx.query;
	const data = {
		comm: { ct: 24 },
		mv_tag: {
			module: "MvService.MvInfoProServer",
			method: "GetAllocTag",
			param: {}
		},
		mv_list: {
			module: "MvService.MvInfoProServer",
			method: "GetAllocMvInfo",
			param: {
				start: (+page ? +page - 1 : 0) * +limit,
				limit: +limit,
				version_id,
				area_id,
				order: 1
			}
		}
	};
	const props = {
		method: "get",
		params: {
			format: "json",
			data: JSON.stringify(data)
		},
		option: {}
	};
	if (!version_id || !area_id) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "version_id or area_id is null" }
		});
		return;
	}
	setApiResponse(ctx, require_services.customResponse({ response: (await require_services.UCommon_default(props)).data }, 200));
});
//#endregion
//#region src/controllers/getMvCategory.ts
var getMvCategory_default = createController(require_services.getMvCategory_default, { name: "getMvCategory" });
//#endregion
//#region src/controllers/getSingerList.ts
var getSingerListController = withErrorHandler(async (ctx) => {
	const { area = -100, sex = -100, genre = -100, index = -100, page = 1 } = ctx.query;
	const pageNum = Number(page);
	const data = {
		comm: {
			ct: 24,
			cv: 0
		},
		singerList: {
			module: "Music.SingerListServer",
			method: "get_singer_list",
			param: {
				area: Number(area),
				sex: Number(sex),
				genre: Number(genre),
				index: Number(index),
				sin: (pageNum - 1) * 80,
				cur_page: pageNum
			}
		}
	};
	setApiResponse(ctx, require_services.customResponse({ response: (await require_services.UCommon_default({
		method: "get",
		params: {
			format: "json",
			data: JSON.stringify(data)
		},
		option: {}
	})).data }, 200));
});
//#endregion
//#region src/controllers/getSingerCategory.ts
var getSingerCategory_default = createController(require_services.getSingerCategory_default, { name: "getSingerCategory" });
//#endregion
//#region src/controllers/getSimilarSinger.ts
var getSimilarSinger_default = createController(require_services.getSimilarSinger_default, {
	validator: validateRequired(["singermid"]),
	errorMessage: "no singermid"
});
//#endregion
//#region src/controllers/getSingerAlbum.ts
var getSingerAlbumController = withErrorHandler(async (ctx) => {
	const singermid = ctx.query.singermid;
	const num = +ctx.query.limit || 5;
	const data = {
		comm: {
			ct: 24,
			cv: 0
		},
		singer: {
			method: "GetAlbumList",
			param: {
				sort: 5,
				singermid,
				begin: +ctx.query.page || 0,
				num
			},
			module: "music.musichallAlbum.AlbumListServer"
		}
	};
	const props = {
		method: "get",
		params: {
			format: "json",
			singermid,
			data: JSON.stringify(data)
		},
		option: {}
	};
	if (!singermid) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "no singermid" }
		});
		return;
	}
	setApiResponse(ctx, require_services.customResponse({ response: (await require_services.UCommon_default(props)).data }, 200));
});
//#endregion
//#region src/controllers/getSingerHotsong.ts
var getSingerHotsongController = withErrorHandler(async (ctx) => {
	const singermid = ctx.query.singermid;
	const num = +ctx.query.limit || 5;
	const data = {
		comm: {
			ct: 24,
			cv: 0
		},
		singer: {
			method: "get_singer_detail_info",
			param: {
				sort: 5,
				singermid,
				sin: ((+ctx.query.page || 0) - 1) * num,
				num
			},
			module: "music.web_singer_info_svr"
		}
	};
	const props = {
		method: "get",
		params: {
			format: "json",
			singermid,
			data: JSON.stringify(data)
		},
		option: {}
	};
	if (!singermid) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "no singermid" }
		});
		return;
	}
	setApiResponse(ctx, require_services.customResponse({ response: (await require_services.UCommon_default(props)).data }, 200));
});
//#endregion
//#region src/controllers/getSingerMv.ts
var getSingerMv_default = withErrorHandler(async (ctx) => {
	const { singermid, order, num = 5 } = ctx.query;
	if (!singermid) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "no singermid" }
		});
		return;
	}
	const orderStr = Array.isArray(order) ? order[0] : order;
	const params = {
		singermid,
		order: orderStr,
		num
	};
	if (orderStr && orderStr.toLowerCase() === "time") params.cmd = 1;
	setApiResponse(ctx, await require_services.getSingerMv_default({
		method: "get",
		params,
		option: {}
	}));
});
//#endregion
//#region src/controllers/getSingerDesc.ts
var getSingerDesc_default = createController(require_services.getSingerDesc_default, {
	validator: validateRequired(["singermid"]),
	errorMessage: "no singermid"
});
//#endregion
//#region src/controllers/getSingerStarNum.ts
var getSingerStarNum_default = createController(require_services.getSingerStarNum_default, {
	validator: validateRequired(["singermid"]),
	errorMessage: "no singermid"
});
//#endregion
//#region src/controllers/getRadioLists.ts
var getRadioLists_default = createController(require_services.getRadioLists_default);
//#endregion
//#region src/controllers/getDigitalAlbumLists.ts
var getDigitalAlbumLists_default = createController(require_services.getDigitalAlbumLists_default);
//#endregion
//#region src/controllers/getLyric.ts
var getLyric_default = withErrorHandler(async (ctx) => {
	const songmid = Array.isArray(ctx.query.songmid) ? ctx.query.songmid[0] : ctx.query.songmid || ctx.params.songmid;
	const rawIsFormat = Array.isArray(ctx.query.isFormat) ? ctx.query.isFormat[0] : ctx.query.isFormat;
	if (!songmid) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "no songmid" }
		});
		return;
	}
	const { cookie: effectiveCookie } = require_cookieResolver.resolveRequestCookie(ctx);
	const headers = {};
	if (effectiveCookie) headers.Cookie = effectiveCookie;
	setApiResponse(ctx, await require_services.getLyric_default({
		method: "get",
		params: { songmid },
		option: { headers },
		isFormat: rawIsFormat
	}));
}, "getLyric");
//#endregion
//#region src/controllers/getMusicPlay.ts
var getMusicPlay_default = withErrorHandler(async (ctx) => {
	const songmid = ctx.query.songmid ?? ctx.params.songmid;
	const resType = Array.isArray(ctx.query.resType) ? ctx.query.resType[0] : ctx.query.resType;
	const mediaId = Array.isArray(ctx.query.mediaId) ? ctx.query.mediaId[0] : ctx.query.mediaId;
	const quality = Array.isArray(ctx.query.quality) ? ctx.query.quality[0] : ctx.query.quality;
	const { cookie: effectiveCookie } = require_cookieResolver.resolveRequestCookie(ctx);
	const headers = {};
	if (effectiveCookie) headers.Cookie = effectiveCookie;
	setApiResponse(ctx, await require_services.getMusicPlay_default({
		method: "get",
		params: {
			songmid,
			resType,
			mediaId,
			quality
		},
		option: { headers }
	}));
});
//#endregion
//#region src/controllers/getAlbumInfo.ts
var getAlbumInfo_default = createController(require_services.getAlbumInfo_default, {
	validator: validateRequired(["albummid"]),
	errorMessage: "no albummid",
	name: "getAlbumInfo"
});
//#endregion
//#region src/controllers/getComments.ts
var getCommentsController = withErrorHandler(async (ctx) => {
	const { id, pagesize = 25, pagenum = 0, cid = 205360772, cmd = 8, reqtype = 2, biztype = 1, rootcommentid = !pagenum && "" } = ctx.query;
	const checkrootcommentid = !pagenum ? true : !!rootcommentid;
	const props = {
		method: "get",
		params: {
			cid,
			reqtype,
			biztype,
			topid: id,
			cmd,
			pagenum,
			pagesize,
			lasthotcommentid: rootcommentid
		},
		option: {}
	};
	if (!id || !checkrootcommentid) {
		setApiResponse(ctx, require_services.errorResponse("Don't have id or rootcommentid", 400));
		return;
	}
	setApiResponse(ctx, await require_services.getComments_default(props));
});
//#endregion
//#region src/controllers/getRecommend.ts
var getRecommend_default = withErrorHandler(async (ctx) => {
	setApiResponse(ctx, {
		status: 200,
		body: { response: (await require_services.UCommon_default({
			method: "get",
			params: {
				format: "json",
				data: JSON.stringify({
					comm: { ct: 24 },
					category: {
						method: "get_hot_category",
						param: { qq: "" },
						module: "music.web_category_svr"
					},
					recomPlaylist: {
						method: "get_hot_recommend",
						param: {
							async: 1,
							cmd: 2
						},
						module: "playlist.HotRecommendServer"
					},
					playlist: {
						method: "get_playlist_by_category",
						param: {
							id: 8,
							curPage: 1,
							size: 40,
							order: 5,
							titleid: 8
						},
						module: "playlist.PlayListPlazaServer"
					},
					new_song: {
						module: "newsong.NewSongServer",
						method: "get_new_song_info",
						param: { type: 5 }
					},
					new_album: {
						module: "newalbum.NewAlbumServer",
						method: "get_new_album_info",
						param: {
							area: 1,
							sin: 0,
							num: 10
						}
					},
					new_album_tag: {
						module: "newalbum.NewAlbumServer",
						method: "get_new_album_area",
						param: {}
					},
					toplist: {
						module: "musicToplist.ToplistInfoServer",
						method: "GetAll",
						param: {}
					},
					focus: {
						module: "QQMusic.MusichallServer",
						method: "GetFocus",
						param: {}
					}
				})
			},
			option: {}
		})).data }
	});
});
//#endregion
//#region src/controllers/getRecommendBanner.ts
var getRecommendBanner_default = createController(require_services.getRecommendBanner_default, { name: "getRecommendBanner" });
//#endregion
//#region src/controllers/getRelatedPlaylists.ts
var getRelatedPlaylists_default = withErrorHandler(async (ctx) => {
	const songid = ctx.query.songid || ctx.query.id;
	if (!songid) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "no songid" }
		});
		return;
	}
	setApiResponse(ctx, await require_services.getRelatedPlaylists_default({
		method: "post",
		params: {
			songid,
			sin: ctx.query.sin,
			lastId: ctx.query.lastId,
			songType: ctx.query.songType
		},
		option: {}
	}));
}, "getRelatedPlaylists");
//#endregion
//#region src/controllers/getRelatedMv.ts
var getRelatedMv_default = withErrorHandler(async (ctx) => {
	const songid = ctx.query.songid || ctx.query.id;
	if (!songid) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "no songid" }
		});
		return;
	}
	setApiResponse(ctx, await require_services.getRelatedMv_default({
		method: "post",
		params: {
			songid,
			songtype: ctx.query.songtype,
			lastmvid: ctx.query.lastmvid,
			limit: ctx.query.limit
		},
		option: {}
	}));
}, "getRelatedMv");
//#endregion
//#region src/controllers/getMvPlay.ts
var getMvPlayController = withErrorHandler(async (ctx) => {
	const { vid } = ctx.query;
	if (!vid) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "vid is null" }
		});
		return;
	}
	const response = await require_services.UCommon_default({
		method: "get",
		params: {
			format: "json",
			data: JSON.stringify({
				comm: {
					ct: 24,
					cv: 4747474
				},
				getMVUrl: {
					module: "gosrf.Stream.MvUrlProxy",
					method: "GetMvUrls",
					param: {
						vids: [vid],
						request_typet: 10001
					}
				},
				mvinfo: {
					module: "video.VideoDataServer",
					method: "get_video_info_batch",
					param: {
						vidlist: [vid],
						required: [
							"vid",
							"type",
							"sid",
							"cover_pic",
							"duration",
							"singers",
							"video_switch",
							"msg",
							"name",
							"desc",
							"playcnt",
							"pubdate",
							"isfav",
							"gmid"
						]
					}
				},
				other: {
					module: "video.VideoLogicServer",
					method: "rec_video_byvid",
					param: {
						vid,
						required: [
							"vid",
							"type",
							"sid",
							"cover_pic",
							"duration",
							"singers",
							"video_switch",
							"msg",
							"name",
							"desc",
							"playcnt",
							"pubdate",
							"isfav",
							"gmid",
							"uploader_headurl",
							"uploader_nick",
							"uploader_encuin",
							"uploader_uin",
							"uploader_hasfollow",
							"uploader_follower_num"
						],
						support: 1
					}
				}
			})
		},
		option: {}
	});
	const mvurls = response.data?.getMVUrl?.data;
	if (!mvurls || typeof mvurls !== "object" || Object.keys(mvurls).length === 0) {
		setApiResponse(ctx, {
			status: 502,
			body: { response: {
				data: null,
				error: "Failed to get MV URL data"
			} }
		});
		return;
	}
	const mvurlskey = Object.keys(mvurls)[0];
	const mp4_urls = mvurls[mvurlskey]?.mp4?.map((item) => item.freeflow_url) || [];
	const hls_urls = mvurls[mvurlskey]?.hls?.map((item) => item.freeflow_url) || [];
	const urls = [...mp4_urls, ...hls_urls];
	let play_urls = [];
	const playLists = {
		f10: [],
		f20: [],
		f30: [],
		f40: []
	};
	if (urls.length) {
		urls.forEach((url) => {
			play_urls = [...play_urls, ...url];
		});
		playLists.f10 = play_urls.filter((item) => /\.f10\.mp4/.test(item));
		playLists.f20 = play_urls.filter((item) => /\.f20\.mp4/.test(item));
		playLists.f30 = play_urls.filter((item) => /\.f30\.mp4/.test(item));
		playLists.f40 = play_urls.filter((item) => /\.f40\.mp4/.test(item));
	}
	response.data.playLists = playLists;
	setApiResponse(ctx, require_services.customResponse({ response: response.data }, 200));
});
//#endregion
//#region src/controllers/getTopLists.ts
var getTopLists_default = createController(require_services.getTopLists_default);
//#endregion
//#region src/controllers/getRanks.ts
var getRanksController = withErrorHandler(async (ctx) => {
	const getWeekNumber = (d) => {
		d = new Date(Date.UTC(d.getFullYear(), d.getMonth(), d.getDate()));
		const dayNum = d.getUTCDay() || 7;
		d.setUTCDate(d.getUTCDate() + 4 - dayNum);
		const yearStart = new Date(Date.UTC(d.getUTCFullYear(), 0, 1));
		return Math.ceil(((d.getTime() - yearStart.getTime()) / 864e5 + 1) / 7);
	};
	const topId = +ctx.query.topId || 4;
	const num = +ctx.query.limit || 20;
	const offset = +ctx.query.page || 0;
	const date = /* @__PURE__ */ new Date();
	const week = getWeekNumber(date);
	const data = {
		comm: {
			ct: 24,
			cv: 4747474,
			format: "json",
			inCharset: "utf-8",
			needNewCode: 1,
			uin: 0
		},
		req_1: {
			module: "musicToplist.ToplistInfoServer",
			method: "GetDetail",
			param: {
				topId,
				offset,
				num,
				period: `${date.getFullYear()}_${week}`
			}
		}
	};
	setApiResponse(ctx, require_services.customResponse({ response: (await require_services.UCommon_default({
		method: "get",
		params: {
			format: "json",
			data: JSON.stringify(data)
		},
		option: {}
	})).data }, 200));
});
//#endregion
//#region src/controllers/getTicketInfo.ts
var getTicketInfo_default = withErrorHandler(async (ctx) => {
	setApiResponse(ctx, {
		status: 200,
		body: { response: (await require_services.UCommon_default({
			method: "get",
			params: {
				format: "json",
				inCharset: "utf8",
				outCharset: "GB2312",
				platform: "yqq.json",
				data: JSON.stringify({
					comm: {
						ct: 24,
						cv: 0
					},
					getFirstData: {
						module: "mall.ticket_index_page_svr",
						method: "GetTicketIndexPage",
						param: { city_id: -1 }
					},
					getTag: {
						module: "mall.ticket_index_page_svr",
						method: "GetShowTypeList",
						param: {}
					}
				})
			},
			option: {}
		})).data }
	});
});
//#endregion
//#region src/controllers/getImageUrl.ts
var getImageUrl_default = withErrorHandler(async (ctx) => {
	const { id, size = "300x300", maxAge = 2592e3 } = ctx.query;
	if (!id) {
		setApiResponse(ctx, {
			status: 400,
			body: { response: "no id~~" }
		});
		return;
	}
	setApiResponse(ctx, {
		status: 200,
		body: { response: {
			code: 0,
			data: { imageUrl: `https://y.gtimg.cn/music/photo_new/T002R${size}M000${id}.jpg?max_age=${maxAge}` }
		} }
	});
});
//#endregion
//#region src/controllers/getQQLoginQr.ts
var getQQLoginQr_default = createController(require_services.getQQLoginQr$1);
//#endregion
//#region src/controllers/checkQQLoginQr.ts
var controller = async (ctx, _next) => {
	const { ptqrtoken, qrsig } = ctx.query.ptqrtoken ? ctx.query : ctx.request.body || {};
	const { status, body } = await require_services.checkQQLoginQr$1({
		method: "get",
		option: {},
		params: {
			ptqrtoken,
			qrsig
		}
	});
	setApiResponse(ctx, {
		status: status || 500,
		body
	});
};
//#endregion
//#region src/controllers/cookies.ts
var cookies_default = {
	get: async (ctx, next) => {
		const userInfo = require_cookieResolver.getUserInfo();
		ctx.status = 200;
		ctx.body = { data: {
			code: 200,
			cookie: userInfo.cookie,
			cookieList: userInfo.cookieList,
			cookieObject: userInfo.cookieObject
		} };
		await next();
	},
	set: async (ctx, next) => {
		ctx.request.cookies = require_cookieResolver.getUserInfo().cookie;
		ctx.request.header["Access-Control-Allow-Origin"] = "https://y.qq.com";
		ctx.request.header["Access-Control-Allow-Methods"] = "GET,PUT,POST,DELETE";
		ctx.request.header["Access-Control-Allow-Headers"] = "Content-Type";
		ctx.request.header["Access-Control-Allow-Credentials"] = true;
		ctx.body = { data: {
			code: 200,
			message: "操作成功"
		} };
		await next();
	}
};
//#endregion
//#region src/controllers/getUserPlaylists.ts
var getUserPlaylists_default = withErrorHandler(async (ctx) => {
	const { uin, offset = 0, limit = 30 } = ctx.query;
	if (!uin) {
		setApiResponse(ctx, {
			status: 400,
			body: { error: "缺少 uin 参数" }
		});
		return;
	}
	const normalizedUin = Array.isArray(uin) ? uin[0] : uin;
	const { cookie } = require_cookieResolver.resolveRequestCookie(ctx);
	setApiResponse(ctx, await require_services.getUserPlaylists$1({
		uin: String(normalizedUin),
		offset: Number(offset),
		limit: Number(limit),
		cookie
	}));
});
//#endregion
//#region src/controllers/getUserAvatar.ts
var getUserAvatarController = withErrorHandler(async (ctx) => {
	const rawK = Array.isArray(ctx.query.k) ? ctx.query.k[0] : ctx.query.k;
	const rawUin = Array.isArray(ctx.query.uin) ? ctx.query.uin[0] : ctx.query.uin;
	const rawSize = Array.isArray(ctx.query.size) ? ctx.query.size[0] : ctx.query.size;
	const parsedSize = rawSize ? Number(rawSize) : 140;
	if (!rawK && !rawUin) {
		setApiResponse(ctx, require_services.errorResponse("缺少 k 或 uin 参数", 400));
		return;
	}
	if (!Number.isFinite(parsedSize) || parsedSize <= 0) {
		setApiResponse(ctx, require_services.errorResponse("size 参数无效", 400));
		return;
	}
	setApiResponse(ctx, require_services.customResponse({ response: {
		code: 0,
		data: {
			avatarUrl: (await require_services.getUserAvatar$1({
				k: rawK,
				uin: rawUin,
				size: parsedSize
			})).avatarUrl,
			message: "获取头像成功"
		}
	} }, 200));
});
//#endregion
//#region src/controllers/getUserLikedSongs.ts
var getUserLikedSongs_default = withErrorHandler(async (ctx) => {
	const { uin, offset = 0, limit = 30 } = ctx.query;
	if (!uin) {
		setApiResponse(ctx, {
			status: 400,
			body: { error: "缺少 uin 参数" }
		});
		return;
	}
	setApiResponse(ctx, await require_services.getUserLikedSongs$1({
		uin,
		offset: Number(offset),
		limit: Number(limit)
	}));
});
//#endregion
//#region src/controllers/getUserReadonlyExtras.ts
var getSingleQueryValue = (value) => {
	if (Array.isArray(value)) return getSingleQueryValue(value[0]);
	if (value === void 0 || value === null) return void 0;
	return String(value).trim() || void 0;
};
var getPaginationValue = (value, fallback) => {
	const rawValue = getSingleQueryValue(value);
	if (!rawValue) return fallback;
	const parsedValue = Number(rawValue);
	return Number.isFinite(parsedValue) ? parsedValue : fallback;
};
var createUserReadonlyController = (service, name) => withErrorHandler(async (ctx) => {
	const uin = getSingleQueryValue(ctx.query.uin) || getSingleQueryValue(ctx.query.id);
	if (!uin) {
		setApiResponse(ctx, {
			status: 400,
			body: { error: "缺少 uin 参数" }
		});
		return;
	}
	const { cookie } = require_cookieResolver.resolveRequestCookie(ctx);
	setApiResponse(ctx, await service({
		uin,
		page: getPaginationValue(ctx.query.page || ctx.query.pageNo, 1),
		limit: getPaginationValue(ctx.query.limit || ctx.query.pageSize, 20),
		cookie
	}));
}, name);
var getUserDetailController = createUserReadonlyController(require_services.getUserDetail$1, "getUserDetail");
var getUserCollectedSongListsController = createUserReadonlyController(require_services.getUserCollectedSongLists$1, "getUserCollectedSongLists");
var getUserCollectedAlbumsController = createUserReadonlyController(require_services.getUserCollectedAlbums$1, "getUserCollectedAlbums");
var getUserFollowSingersController = createUserReadonlyController(require_services.getUserFollowSingers$1, "getUserFollowSingers");
var getUserFollowUsersController = createUserReadonlyController(require_services.getUserFollowUsers$1, "getUserFollowUsers");
var getUserFansController = createUserReadonlyController(require_services.getUserFans$1, "getUserFans");
//#endregion
//#region src/services/apis/recommend/getDailyRecommend.ts
var DEFAULT_NEW_SONG_TYPE = 5;
function normalizeNewSongType(areaId) {
	return Number.isInteger(areaId) && areaId >= 1 && areaId <= 5 ? areaId : DEFAULT_NEW_SONG_TYPE;
}
/**
* 获取每日推荐歌曲
* @param cookie - 用户 Cookie（可选，用于获取个性化推荐）
* @returns 每日推荐歌曲列表
*/
async function getDailyRecommend(cookie) {
	const data = {
		comm: {
			...require_services.getCommonParams(),
			ct: 24,
			cv: 0
		},
		recommend: {
			method: "get_recommend",
			module: "music.web_srf_svr",
			param: {
				page: 0,
				num: 20,
				uin: 0,
				login: 0
			}
		}
	};
	const headers = { "Content-Type": "application/json" };
	if (cookie) headers.cookie = cookie;
	return require_services.handleApi(require_services.u_common_default({
		method: "POST",
		options: {
			data: JSON.stringify(data),
			headers
		}
	}));
}
/**
* 获取私人 FM
* @param cookie - 用户 Cookie
* @returns 私人 FM 歌曲列表
*/
async function getPrivateFM(cookie) {
	const data = {
		comm: {
			...require_services.getCommonParams(),
			ct: 24,
			cv: 0
		},
		fm: {
			method: "GetPrivateFmPlaylist",
			module: "music.web_fm_svr",
			param: {
				enc: "utf8",
				moter: 0,
				uin: 0,
				login: 0
			}
		}
	};
	const headers = { "Content-Type": "application/json" };
	if (cookie) headers.cookie = cookie;
	return require_services.handleApi(require_services.u_common_default({
		method: "POST",
		options: {
			data: JSON.stringify(data),
			headers
		}
	}));
}
/**
* 获取新歌速递
* @param areaId - 地区 ID（兼容旧参数，当前映射到新歌类型，暂定映射关系）
* @param limit - 返回数量（当前上游接口固定类型，limit 仅用于兼容入参）
* @returns 新歌列表
*/
async function getNewSongs(areaId = DEFAULT_NEW_SONG_TYPE, limit = 20) {
	const normalizedType = normalizeNewSongType(areaId);
	const normalizedLimit = Number.isFinite(limit) && limit > 0 ? Math.floor(limit) : 20;
	const data = {
		comm: {
			...require_services.getCommonParams(),
			ct: 24,
			cv: 0
		},
		new_song: {
			module: "newsong.NewSongServer",
			method: "get_new_song_info",
			param: {
				type: normalizedType,
				num: normalizedLimit
			}
		}
	};
	return require_services.handleApi(require_services.u_common_default({
		method: "POST",
		options: {
			data: JSON.stringify(data),
			headers: { "Content-Type": "application/json" }
		}
	}));
}
var getDailyRecommend_default = {
	getDailyRecommend,
	getPrivateFM,
	getNewSongs
};
//#endregion
//#region src/controllers/getDailyRecommend.ts
async function getDailyRecommendController(ctx) {
	const { cookie } = require_cookieResolver.resolveRequestCookie(ctx);
	setApiResponse(ctx, await getDailyRecommend_default.getDailyRecommend(cookie));
}
async function getPrivateFMController(ctx) {
	const { cookie } = require_cookieResolver.resolveRequestCookie(ctx);
	setApiResponse(ctx, await getDailyRecommend_default.getPrivateFM(cookie));
}
async function getNewSongsController(ctx) {
	const { areaId = "5", limit = "20" } = ctx.query;
	setApiResponse(ctx, await getDailyRecommend_default.getNewSongs(Number(areaId), Number(limit)));
}
//#endregion
//#region src/services/apis/recommend/getPersonalRecommend.ts
var PERSONAL_RECOMMEND_CONFIG = {
	1: {
		key: "recomPlaylist",
		method: "get_recommend",
		module: "music.web_srf_svr",
		param: {
			page: 0,
			num: 20,
			uin: 0,
			login: 0
		}
	},
	2: {
		key: "radio",
		method: "get_radio_list",
		module: "pf.radiosvr",
		param: {
			page_offset: 1,
			page_size: 20
		}
	},
	3: {
		key: "mv",
		method: "GetRecommendMV",
		module: "gosrf.Stream.MvUrlProxy",
		param: { size: 20 }
	}
};
var DEFAULT_PERSONAL_RECOMMEND_TYPE = 1;
var DEFAULT_SIMILAR_SONG_LIMIT = 20;
function resolveRecommendConfig(type) {
	return PERSONAL_RECOMMEND_CONFIG[type] || PERSONAL_RECOMMEND_CONFIG[DEFAULT_PERSONAL_RECOMMEND_TYPE];
}
/**
* 获取个性化推荐
* @param type - 推荐类型 (1: 歌单，2: 电台，3: MV)
* @param cookie - 用户 Cookie
* @returns 推荐列表
*/
async function getPersonalRecommend(type = 1, cookie) {
	const recommendConfig = resolveRecommendConfig(type);
	const data = {
		comm: {
			...require_services.getCommonParams(),
			ct: 24,
			cv: 0
		},
		[recommendConfig.key]: {
			method: recommendConfig.method,
			module: recommendConfig.module,
			param: recommendConfig.param
		}
	};
	const headers = { "Content-Type": "application/json" };
	if (cookie) headers.cookie = cookie;
	return require_services.handleApi(require_services.u_common_default({
		method: "POST",
		options: {
			data: JSON.stringify(data),
			headers
		}
	}));
}
/**
* 获取相似歌曲推荐
* @param songmid - 歌曲 MID
* @param cookie - 用户 Cookie
* @returns 相似歌曲列表
*/
async function getSimilarSongs(songmid, cookie) {
	const data = {
		comm: {
			...require_services.getCommonParams(),
			ct: 24,
			cv: 0
		},
		similarSong: {
			method: "get_similar_song_info",
			module: "music.web_srf_svr",
			param: {
				songid: 0,
				songmid,
				num: DEFAULT_SIMILAR_SONG_LIMIT
			}
		}
	};
	const headers = { "Content-Type": "application/json" };
	if (cookie) headers.cookie = cookie;
	return require_services.handleApi(require_services.u_common_default({
		method: "POST",
		options: {
			data: JSON.stringify(data),
			headers
		}
	}));
}
var getPersonalRecommend_default = {
	getPersonalRecommend,
	getSimilarSongs
};
//#endregion
//#region src/controllers/getPersonalRecommend.ts
async function getPersonalRecommendController(ctx) {
	const { type = "1" } = ctx.query;
	const rawType = Array.isArray(type) ? type[0] : type;
	const { cookie } = require_cookieResolver.resolveRequestCookie(ctx);
	setApiResponse(ctx, await getPersonalRecommend_default.getPersonalRecommend(Number(rawType), cookie));
}
async function getSimilarSongsController(ctx) {
	const { songmid } = ctx.query;
	if (!songmid) {
		setApiResponse(ctx, require_services.errorResponse("缺少参数 songmid", 400));
		return;
	}
	const validSongmid = Array.isArray(songmid) ? songmid[0] : songmid;
	if (!validSongmid || String(validSongmid).trim() === "") {
		setApiResponse(ctx, require_services.errorResponse("参数 songmid 不能为空", 400));
		return;
	}
	const { cookie } = require_cookieResolver.resolveRequestCookie(ctx);
	setApiResponse(ctx, await getPersonalRecommend_default.getSimilarSongs(String(validSongmid), cookie));
}
//#endregion
//#region src/services/apis/extend/getPlaylistTags.ts
/**
* 获取歌单标签列表
* @returns 歌单标签列表
*/
async function getPlaylistTags() {
	const data = {
		comm: {
			...require_services.getCommonParams(),
			ct: 24,
			cv: 0
		},
		playlist: {
			method: "get_tags",
			module: "playlist.web_srf",
			param: {}
		}
	};
	return require_services.handleApi(require_services.u_common_default({
		method: "POST",
		options: {
			data: JSON.stringify(data),
			headers: { "Content-Type": "application/json" }
		}
	}));
}
/**
* 根据标签获取歌单列表
* @param tagId - 标签 ID
* @param page - 页码
* @param num - 每页数量
* @returns 歌单列表
*/
async function getPlaylistsByTag(tagId = 1, page = 0, num = 20) {
	const data = {
		comm: {
			...require_services.getCommonParams(),
			ct: 24,
			cv: 0
		},
		playlist: {
			method: "get_playlist_by_tag",
			module: "playlist.web_srf",
			param: {
				tag_id: tagId,
				rec_type: 3,
				show_list: 1,
				sort: 1,
				user_id: 0,
				page,
				num
			}
		}
	};
	return require_services.handleApi(require_services.u_common_default({
		method: "POST",
		options: {
			data: JSON.stringify(data),
			headers: { "Content-Type": "application/json" }
		}
	}));
}
/**
* 获取热门评论
* @param id - 资源 ID（歌曲/歌单/专辑 ID）
* @param type - 资源类型 (1: 歌曲，2: 歌单，3: 专辑)
* @param page - 页码
* @param pagesize - 每页数量
* @returns 评论列表
*/
async function getHotComments(id, type = 1, page = 0, pagesize = 20) {
	const data = {
		comm: {
			...require_services.getCommonParams(),
			ct: 24,
			cv: 0
		},
		comment: {
			module: "comment.CommentReadServer",
			method: "GetCommentList",
			param: {
				biztype: type,
				bizid: id,
				page,
				pagesize,
				needhot: 1
			}
		}
	};
	return require_services.handleApi(require_services.u_common_default({
		method: "POST",
		options: {
			data: JSON.stringify(data),
			headers: { "Content-Type": "application/json" }
		}
	}));
}
/**
* 获取歌手分类列表
* @param area - 地区 (1: 内地，2: 港台，3: 欧美，4: 韩国，5: 日本)
* @param sex - 性别 (1: 男歌手，2: 女歌手，3: 组合)
* @param genre - 流派 (1: 流行，2: 摇滚，3: 民谣等)
* @param page - 页码
* @param pagesize - 每页数量
* @returns 歌手列表
*/
async function getSingerListByArea(area = -1, sex = -1, genre = -1, page = 1, pagesize = 80) {
	const data = {
		comm: {
			...require_services.getCommonParams(),
			ct: 24,
			cv: 0
		},
		singer: {
			method: "get_singer_list",
			module: "music.web_srf_svr",
			param: {
				area,
				sex,
				genre,
				page,
				pagesize
			}
		}
	};
	return require_services.handleApi(require_services.u_common_default({
		method: "POST",
		options: {
			data: JSON.stringify(data),
			headers: { "Content-Type": "application/json" }
		}
	}));
}
var getPlaylistTags_default = {
	getPlaylistTags,
	getPlaylistsByTag,
	getHotComments,
	getSingerListByArea
};
//#endregion
//#region src/controllers/getPlaylistTags.ts
/**
* 获取歌单标签列表
*/
var getPlaylistTagsController = withErrorHandler(async (ctx) => {
	setApiResponse(ctx, await getPlaylistTags_default.getPlaylistTags());
});
/**
* 根据标签获取歌单列表
*/
var getPlaylistsByTagController = withErrorHandler(async (ctx) => {
	const { tagId = "1", page = "0", num = "20" } = ctx.query;
	setApiResponse(ctx, await getPlaylistTags_default.getPlaylistsByTag(Number(tagId), Number(page), Number(num)));
});
/**
* 获取热门评论
*/
var getHotCommentsController = withErrorHandler(async (ctx) => {
	const { id, type = "1", page = "0", pagesize = "20" } = ctx.query;
	if (!id) {
		setApiResponse(ctx, require_services.errorResponse("缺少参数 id（资源 ID）", 400));
		return;
	}
	setApiResponse(ctx, await getPlaylistTags_default.getHotComments(id, Number(type), Number(page), Number(pagesize)));
});
/**
* 获取歌手分类列表
*/
var getSingerListByAreaController = withErrorHandler(async (ctx) => {
	const { area = "-1", sex = "-1", genre = "-1", page = "1", pagesize = "80" } = ctx.query;
	setApiResponse(ctx, await getPlaylistTags_default.getSingerListByArea(Number(area), Number(sex), Number(genre), Number(page), Number(pagesize)));
});
//#endregion
//#region src/controllers/index.ts
var controllers$1 = {
	getCookie: cookies_default.get,
	setCookie: cookies_default.set,
	getDownloadQQMusic: getDownloadQQMusic_default,
	getHotKey: getHotkey_default,
	getSearchByKey: getSearchByKey_default,
	getSmartbox: getSmartbox_default,
	getSongListCategories: getSongListCategoriesController,
	getSongLists: getSongLists_default,
	batchGetSongLists: batchGetSongListsController,
	getSongInfo: getSongInfoController,
	batchGetSongInfo: batchGetSongInfoController,
	getSongListDetail: getSongListDetailController,
	getAlbumSongs: getAlbumSongs_default,
	getNewDisks: getNewDisks_default,
	getMvByTag: getMvByTag_default,
	getMv: getMvController,
	getMvCategory: getMvCategory_default,
	getSingerList: getSingerListController,
	getSingerCategory: getSingerCategory_default,
	getSimilarSinger: getSimilarSinger_default,
	getSingerAlbum: getSingerAlbumController,
	getSingerHotsong: getSingerHotsongController,
	getSingerMv: getSingerMv_default,
	getSingerDesc: getSingerDesc_default,
	getSingerStarNum: getSingerStarNum_default,
	getRadioLists: getRadioLists_default,
	getDigitalAlbumLists: getDigitalAlbumLists_default,
	getLyric: getLyric_default,
	getMusicPlay: getMusicPlay_default,
	getAlbumInfo: getAlbumInfo_default,
	getComments: getCommentsController,
	getRecommend: getRecommend_default,
	getRecommendBanner: getRecommendBanner_default,
	getRelatedPlaylists: getRelatedPlaylists_default,
	getRelatedMv: getRelatedMv_default,
	getMvPlay: getMvPlayController,
	getTopLists: getTopLists_default,
	getRanks: getRanksController,
	getTicketInfo: getTicketInfo_default,
	getImageUrl: getImageUrl_default,
	getQQLoginQr: getQQLoginQr_default,
	checkQQLoginQr: controller,
	getUserPlaylists: getUserPlaylists_default,
	getUserAvatar: getUserAvatarController,
	getUserLikedSongs: getUserLikedSongs_default,
	getUserDetail: getUserDetailController,
	getUserCollectedSongLists: getUserCollectedSongListsController,
	getUserCollectedAlbums: getUserCollectedAlbumsController,
	getUserFollowSingers: getUserFollowSingersController,
	getUserFollowUsers: getUserFollowUsersController,
	getUserFans: getUserFansController,
	getDailyRecommend: getDailyRecommendController,
	getPrivateFM: getPrivateFMController,
	getNewSongs: getNewSongsController,
	getPersonalRecommend: getPersonalRecommendController,
	getSimilarSongs: getSimilarSongsController,
	getPlaylistTags: getPlaylistTagsController,
	getPlaylistsByTag: getPlaylistsByTagController,
	getHotComments: getHotCommentsController,
	getSingerListByArea: getSingerListByAreaController
};
//#endregion
//#region src/routes/router.ts
var router = new _koa_router.Router();
var controllers = controllers$1;
var getPaths = (item) => [item.path, ...item.aliases || []];
var registerRoute = (method, path, controller) => {
	switch (method) {
		case "GET":
			router.get(path, controller);
			return;
		case "POST":
			router.post(path, controller);
			return;
		case "DELETE":
			router.delete(path, controller);
			return;
		default: throw new Error(`Unsupported API method: ${method}`);
	}
};
apiMetadata.forEach((item) => {
	const controller = controllers[item.name];
	if (!controller) throw new Error(`Missing controller for API metadata item: ${item.name}`);
	getPaths(item).forEach((path) => registerRoute(item.method, path, controller));
});
//#endregion
//#region src/util/cookie.ts
var SAFE_COOKIE_NAMES = new Set(["qqmusic_key", "qqmusic_uin"]);
var cookieMiddleware = (options = {}) => async (ctx, next) => {
	const useFallback = options.fallbackMode ?? require_cookieResolver.service_config_default.fallbackMode;
	const useGlobal = options.useGlobalCookie ?? require_cookieResolver.service_config_default.useGlobalCookie;
	const { cookie } = require_cookieResolver.resolveRequestCookie(ctx, {
		fallbackMode: useFallback,
		useGlobalCookie: useGlobal,
		cookieParamName: options.cookieParamName ?? require_cookieResolver.service_config_default.cookieParamName
	});
	if (cookie) require_cookieResolver.setRequestCookieContext(ctx, cookie);
	const userInfo = require_cookieResolver.getUserInfo();
	if (useGlobal && Array.isArray(userInfo.cookieList)) userInfo.cookieList.forEach((cookieItem) => {
		const [key, ...valueParts] = cookieItem.split("=");
		const normalizedKey = key?.trim();
		const value = valueParts.join("=").trim();
		if (normalizedKey && value && SAFE_COOKIE_NAMES.has(normalizedKey)) ctx.cookies.set(normalizedKey, value, {
			overwrite: true,
			httpOnly: false,
			sameSite: "lax"
		});
	});
	await next();
};
//#endregion
//#region src/koaApp.ts
var __dirname$1 = node_path.default.dirname((0, node_url.fileURLToPath)(require("url").pathToFileURL(__filename).href));
var app = new koa.default();
var publicDir = fs.default.existsSync(node_path.default.join(__dirname$1, "public")) ? node_path.default.join(__dirname$1, "public") : node_path.default.join(__dirname$1, "..", "public");
require_cookieResolver.initializeUserInfo();
app.use(async (ctx, next) => {
	try {
		await next();
	} catch (err) {
		const e = err;
		ctx.status = e.status || 500;
		ctx.type = "application/json";
		ctx.body = { error: e.status && e.status < 500 ? e.message || "请求处理失败" : "服务器内部错误" };
		ctx.app.emit("error", err, ctx);
	}
});
app.on("error", (err, ctx) => {
	const e = err;
	const status = e.status || 500;
	const line = `${ctx?.method || "?"} ${ctx?.url || "?"} → ${status} ${e.message || "Unknown"}`;
	if (status >= 500) console.error(require_services.colors.error(line));
	else console.log(require_services.colors.info(line));
});
app.use(async (ctx, next) => {
	await next();
	const rt = ctx.response.get("X-Response-Time");
	console.log(require_services.colors.prompt(`${ctx.method} ${ctx.url} - ${rt}`));
});
app.use(async (ctx, next) => {
	const start = Date.now();
	await next();
	const ms = Date.now() - start;
	ctx.set("X-Response-Time", `${ms}ms`);
});
app.use(crossOrigin({
	exposeHeaders: ["WWW-Authenticate", "Server-Authorization"],
	maxAge: 5,
	allowMethods: [
		"GET",
		"POST",
		"DELETE"
	],
	allowHeaders: [
		"Content-Type",
		"Authorization",
		"Accept"
	]
}));
app.use(securityHeaders());
app.use(bodyParser());
app.use(cookieMiddleware());
app.use(apiExplorer({ publicDir }));
app.use(staticServe(publicDir));
app.use(router.routes());
app.use(router.allowedMethods());
//#endregion
//#region src/server.ts
var server_exports = /* @__PURE__ */ require_cookieResolver.__exportAll({ startServer: () => startServer });
var parsedPort = Number.parseInt(process.env.PORT ?? "", 10);
var PORT = Number.isFinite(parsedPort) ? parsedPort : 3200;
var startServer = (options = {}) => {
	const port = options.port ?? PORT;
	if (!options.json) {
		console.log((0, node_util.styleText)("green", "\n🎵 QQ Music API Service Starting...\n"));
		console.log(require_services.colors.info(`Current Version: ${require_package.package_default.version}`));
		console.log(require_services.colors.info(`Fallback Mode: ${require_cookieResolver.service_config_default.fallbackMode ? "Enabled" : "Disabled"}`));
		console.log(require_services.colors.info(`Use Global Cookie: ${require_cookieResolver.service_config_default.useGlobalCookie ? "Yes" : "No"}`));
		if (require_cookieResolver.service_config_default.fallbackMode) {
			console.log((0, node_util.styleText)("green", "\n✅ 降级模式已启用：支持手动传递 Cookie\n"));
			console.log("使用方式:");
			console.log("  1. Query 参数：GET /api/endpoint?cookie=your_cookie");
			console.log("  2. Header: X-Custom-Cookie: your_cookie");
			console.log("  3. Header: Cookie: your_cookie\n");
		}
		if (!require_cookieResolver.service_config_default.useGlobalCookie) console.log((0, node_util.styleText)("yellow", "\n⚠️  全局 Cookie 未启用：需要登录的接口请手动传递 Cookie\n"));
		else {
			console.log((0, node_util.styleText)("green", "\n✅ 全局 Cookie 已启用\n"));
			const userInfo = require_cookieResolver.getUserInfo();
			if (!(userInfo.loginUin || userInfo.uin)) console.log((0, node_util.styleText)("yellow", `😔 The configuration ${(0, node_util.styleText)("red", "loginUin")} or your ${(0, node_util.styleText)("red", "cookie")} in file ${(0, node_util.styleText)("green", "config/user-info")} has not configured. \n`));
			if (!userInfo.cookie) console.log((0, node_util.styleText)("yellow", `😔 The configuration ${(0, node_util.styleText)("red", "cookie")} in file ${(0, node_util.styleText)("green", "config/user-info")} has not configured. \n`));
		}
	}
	const server = app.listen(port, () => {
		if (options.json) {
			console.log(JSON.stringify({
				ok: true,
				command: "serve",
				port
			}));
			return;
		}
		console.log(require_services.colors.prompt(`server running @ http://localhost:${port}`));
		console.log(require_services.colors.info(`open playground @ http://localhost:${port}/index.html`));
		console.log(require_services.colors.info(`open explorer @ http://localhost:${port}/explorer`));
	});
	server.on("error", (error) => {
		if (error.code === "EADDRINUSE") {
			console.error(require_services.colors.error(`Port ${port} is already in use.`));
			console.error(require_services.colors.warn("Stop the existing process or set PORT to another value before running again."));
			process.exit(1);
		}
		console.error(require_services.colors.error("Failed to start server."));
		console.error(error);
		process.exit(1);
	});
	return server;
};
//#endregion
Object.defineProperty(exports, "app", {
	enumerable: true,
	get: function() {
		return app;
	}
});
Object.defineProperty(exports, "server_exports", {
	enumerable: true,
	get: function() {
		return server_exports;
	}
});
Object.defineProperty(exports, "startServer", {
	enumerable: true,
	get: function() {
		return startServer;
	}
});
